---
title: "Metagenomic Profiles of Archaea and Bacteria within Thermal and Geochemical Gradients of the Guaymas Basin Deep Subsurface"
author: David Geller-McGrath
date: May 30, 2023
output:
  html_document:
    df_print: paged
    toc: true # create a table of contents
    toc_depth: 5 # up to five depths of headings in the toc
    theme: united # select the document theme
    syntax: tango # select syntax highlight theme
---

<br>

This code was used for the analyses of Guaymas Basin metagenomes and metatranscriptomes for the 2023 publication "Metagenomic Profiles of Archaea and Bacteria within Thermal and Geochemical Gradients of the Guaymas Basin Deep Subsurface" by Geller-McGrath et al. Some sections of these analyses were done on a compute node of the Poseidon High Performance Computing (HPC) cluster based at the Woods Hole Oceanographic Institute (WHOI). All of the analyses done here using R can be run locally; HPC processing was used to speed up computations.

<br>

Load required R libraries

```{r include=TRUE, message=FALSE, warning=FALSE}
library(tidyverse)
library(vroom)
library(ComplexHeatmap)
library(gridtext)
library(RColorBrewer)
library(ecodist)
library(vegan)
library(ape)
library(ggrepel)
library(latex2exp)
```

<br>

## Format MAG bins from DAS Tool and MAG complenetess/contamination estimates from CheckM2

```{r message=FALSE, warning=FALSE}
library(tidyverse)

dasTool_summary_dec_2022 = read_tsv('~/Downloads/Guaymas_MAG_P__DASTool_summary.tsv', show_col_types = FALSE)


checkm2_andreas_bins = read_tsv('flat_files/andreas_mags_checkm2_quality_report.tsv', show_col_types = FALSE) |>
  rename_with(~ str_to_lower(.x)) |>
  select(-`translation_table used`)


gtdb_andreas_bins = read_tsv('flat_files/andreas_gtdbtk_archaea.tsv', show_col_types = FALSE) |>
  bind_rows(read_tsv('flat_files/andreas_gtdbtk_bacteria.tsv', show_col_types = FALSE)) |>
  select(user_genome, classification) |>
  rename(mag_name = user_genome)


taxonomic_ranks = c("domain", "phylum", "class", "order", "family", "genus", "species")


gtdb_andreas_bins = gtdb_andreas_bins |>
  separate(classification, into = taxonomic_ranks, sep = ';') |>
  mutate(across(all_of(taxonomic_ranks), ~ str_replace(.x, '^[:alpha:]__', ''))) |>
  mutate(across(all_of(taxonomic_ranks), ~ if_else(.x == '', NA_character_, .x))) |>
  arrange(domain, phylum)


andreas_mag_summary_dec_2022 = dasTool_summary_dec_2022 |>
  left_join(checkm2_andreas_bins |> rename(bin = name), by = 'bin') |>
  left_join(gtdb_andreas_bins |> rename(bin = mag_name), by = 'bin') |>
  relocate(c(completeness, contamination, bin_score,
             taxonomic_ranks, SCG_completeness, SCG_redundancy),
           .before = bin)


# 125 MAGs that are not contaminants
# 89 that are >= 50% completeness, <= 10% contamination
final_filtered_andreas_mag_summary_dec_2022 = andreas_mag_summary_dec_2022 |>
  filter(
    !phylum %in% c('Firmicutes', 'Actinobacteriota', 'Patescibacteria'), # remove contaminant MAGs
    !class %in% c('Gammaproteobacteria'), # remove contaminant MAGs
    completeness >= 50,
    contamination <= 10)


final_filtered_andreas_mag_summary_dec_2022 = final_filtered_andreas_mag_summary_dec_2022 |>
  arrange(desc(completeness),  desc(contamination)) |>
  mutate(mag_name = paste0(
    'Guaymas_MAG_P_',
    1:nrow(final_filtered_andreas_mag_summary_dec_2022) |> str_pad(width = 3, side = 'left', pad = '0')),
    .before = 1)


#13 unique bacterial phyla recovered
final_filtered_andreas_mag_summary_dec_2022 |>
  filter(domain == 'Bacteria') |>
  pull(phylum) |>
  unique() |>
  length()


#6 unique archaeal phyla recovered
final_filtered_andreas_mag_summary_dec_2022 |>
  filter(domain == 'Archaea') |>
  pull(phylum) |>
  unique() |>
  length()
```



## Figure 2. MAG relative abundances
```{r echo=TRUE, message=FALSE, warning=FALSE}
# Figure 2
final_filtered_andreas_mag_summary_dec_2022 =
  readRDS('rds_files/final_filtered_andreas_mag_summary_dec_2022.rds')

env_metadata3 = readRDS('rds_files/env_metadata3.rds')


guay_relAbun_files = system('ls flat_files/mag_relative_abundance_coverM/new_rel_abun_dec_2022/*.tsv',
                            intern = TRUE)


guay_relAbun_tbl = guay_relAbun_files |>
  map(~ read_tsv(.x, show_col_types = FALSE)) |>
  reduce(left_join, by = 'Genome')


guay_relAbun_tbl = guay_relAbun_tbl |>
  rename_with(~ str_replace(.x, '(.*)_(.*)_S\\d+.*', '\\1\\2')) |>
  rename_with(~ str_replace(.x, '(.*)-(.*)_S\\d+.*', '\\1_\\2')) |>
  rename_with(~ str_replace(.x, '^U(.*B)(.*)', '\\1_\\2')) |>
  rename(bin = Genome) |>
  slice(-1) |> # remove the unmapped % information; won't be used in the heatmap
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, bin),
            by = 'bin') |>
  relocate(mag_name, .before = 1) |>
  filter(!is.na(mag_name)) |>
  select(-bin)


chloroflexota_percent_abundance = guay_relAbun_tbl |> 
  left_join(final_filtered_andreas_mag_summary_dec_2022 |> 
              select(mag_name, phylum)) |> 
  filter(phylum == 'Chloroflexota') |> 
  select(-phylum) |>
  summarize(across(2:last_col(), ~ sum(.x))) |> 
  pivot_longer(
    cols = 2:last_col(),
    names_to = 'sample',
    values_to = 'chloroflexota_percent_abundance'
  )


guayMagRelPalette = list(
  `Phylum/Class` = c(
    'Candidatus Niteriota'	=	'wheat1',
    'Acidobacteriota'	=	'tomato',
    'Aenigmatarchaeota'	=	'orchid1',
    'Aerophobota'	=	'thistle1',
    'Alphaproteobacteria'	=	'slateblue1',
    'Asgardarchaeota'	=	'aquamarine',
    'Atribacterota'	=	'gold',
    'Bipolaricaulota'	=	'darkred',
    'Chloroflexota'	=	'darkviolet',
    'Cloacimonadota'	=	'khaki',
    'Desulfobacterota'	=	'firebrick1',
    'Hadarchaeota'	=	'gainsboro',
    'Marinisomatota'	=	'springgreen',
    'Methanobacteriota_B'	=	'chocolate4',
    'Planctomycetota'	=	'yellowgreen',
    'Candidatus Rosutiota'	=	'deepskyblue',
    'Thermoplasmatota'	=	'#666666',
    'Thermoproteota'	=	'#B89B74',
    'Unclassified'	=	'lightsteelblue1',
    'WOR-3'	=	'ivory',
    "Zixibacteria"	=	'#2A7FB7'),
  
  `Depth (mbsf)` = c('white', 'black'),
  
  Site = brewer.pal(8, 'Pastel2')[3:7] |>
    set_names('U1545B', 'U1546B', 'U1547B', 'U1548B', 'U1549B'))

guay_relAbun_plot_data =
  guay_relAbun_tbl |>
  mutate(across(2:last_col(), ~ log1p(.x))) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, domain, phylum, class),
            by = 'mag_name') |>
  mutate(`Phylum/Class` = case_when(
    class == 'Alphaproteobacteria' ~ 'Alphaproteobacteria',
    is.na(phylum) ~ 'Unclassified',
    TRUE ~ phylum)) |>
  select(-c(class, phylum)) |>
  relocate(c(domain, `Phylum/Class`), .after = 1) |>
  group_by(`Phylum/Class`) |>
  mutate(group_total = n()) |>
  ungroup() |>
  arrange(factor(domain, levels = c('Bacteria', 'Archaea')), 
          desc(group_total), `Phylum/Class`) |>
  select(-c(domain, `Phylum/Class`, group_total)) |>
  column_to_rownames('mag_name')


guayMagRelAbun_rowAnno = guay_relAbun_tbl |>
  select(mag_name) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, domain, phylum, class),
            by = 'mag_name') |>
  mutate(`Phylum/Class` = case_when(
    class == 'Alphaproteobacteria' ~ 'Alphaproteobacteria',
    is.na(phylum) ~ 'Unclassified',
    TRUE ~ phylum)) |>
  select(-c(class, phylum)) |>
  column_to_rownames('mag_name')



guay_row_anno = guay_relAbun_tbl |>
  mutate(across(2:last_col(), ~ log1p(.x))) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, domain, phylum, class),
            by = 'mag_name') |>
  mutate(`Phylum/Class` = case_when(
    class == 'Alphaproteobacteria' ~ 'Alphaproteobacteria',
    is.na(phylum) ~ 'Unclassified',
    TRUE ~ phylum)) |>
  select(-c(class, phylum)) |>
  relocate(c(domain, `Phylum/Class`), .after = 1) |>
  group_by(`Phylum/Class`) |>
  mutate(group_total = n()) |>
  ungroup() |>
  arrange(factor(domain, levels = c('Bacteria', 'Archaea')), 
          desc(group_total), `Phylum/Class`) |>
  select(mag_name, `Phylum/Class`) |>
  mutate(`Phylum/Class` = case_when(
    `Phylum/Class` == 'TA06_A' ~ 'Candidatus Rosutiota',
    `Phylum/Class` == '4484-113' ~ 'Candidatus Niteriota',
    TRUE ~ `Phylum/Class`),
    `Phylum/Class` = fct_inorder(`Phylum/Class`)) |>
  column_to_rownames('mag_name')


metagenome_sample_metadata = read_tsv('flat_files/metagenome_sample_name_metadata.tsv')


guay_col_anno = metagenome_sample_metadata |>
  mutate(Sample = paste0('U', Sample)) |>
  left_join(env_metadata3 |>
              select(-c(`Depth (mbsf)`)),
            by = c('Sample' = 'sample')) |>
  column_to_rownames('Sample')


guay_relAbun_plot_data = guay_relAbun_plot_data |>
  rename_with(~ paste0('U', .x), .cols = everything()) |> 
  rownames_to_column('mag_name') |> 
  relocate(c(mag_name, env_metadata3$sample))


guayMagRelPalette$`Temperature (°C)` = c('white', 'firebrick3')


heat_figure2_new = tibble(
  col = colnames(guay_relAbun_plot_data |> 
                   column_to_rownames('mag_name') |>
                   #rename_with(~ paste0('U', .x)) |>
  relocate(guay_col_anno |>
             select(`Depth (mbsf)`, `T`) |>
             rename(`Temperature (°C)` = `T`) |>
             arrange(`Temperature (°C)`) |>
             rownames()))) |> 
  left_join(env_metadata3 |> 
              select(1:3) |> 
              rename(col = sample))

heat_figure2_new = heat_figure2_new |> 
  select(-`T`) |> 
  rename(depth = 2) |> 
  mutate(col_names = col |> str_replace("(.*_).*", paste0("\\1", depth))) |> 
  select(-depth)
 

guay_relAbun_complexHeatmap =
  guay_relAbun_plot_data |>
  column_to_rownames('mag_name') |>
  #rename_with(~ paste0('U', .x)) |>
  relocate(guay_col_anno |>
             select(`Depth (mbsf)`, `T`) |>
             rename(`Temperature (°C)` = `T`) |>
             arrange(`Temperature (°C)`) |>
             rownames()) |>
  rename_with(~ heat_figure2_new$col_names, .cols = everything()) |> 
  as.matrix() |>
  ComplexHeatmap::pheatmap(
    name = 'Relative abundance
(% total reads mapped)',
main = 'Guaymas MAG relative abundance',
show_colnames = TRUE,
annotation_row = guay_row_anno,
annotation_col = guay_col_anno |>
  select(`Depth (mbsf)`, `T`) |>
  rename(`Temperature (°C)` = `T`) |>
  arrange(`Temperature (°C)`),
annotation_colors = guayMagRelPalette,
cluster_rows = FALSE,
cluster_cols = FALSE,
show_rownames = FALSE,
gaps_row = 63,
gaps_col = c(13, 21),
border_color = 'grey30',
treeheight_row = 0,
treeheight_col = 10,
color = c('grey60', rev(viridis::magma(n = 50))),
legend_breaks = 0:3,
legend_labels = gt_render(c('0', sprintf('%.1f', round(expm1(1:3), 1)))),
show_row_dend = FALSE,
show_column_dend = TRUE)


draw(guay_relAbun_complexHeatmap,
     merge_legend = TRUE)

# pdf(
#   file = '~/Documents/guaymas_analysis/may_25_all_manuscript_docs/figure_2.pdf',
#   width = 7,
#   height = 9)
# draw(guay_relAbun_complexHeatmap,
#      merge_legend = TRUE)
# dev.off()

```

<br> 

## Figure 3. MAG transcript abundances
```{r echo=TRUE, message=FALSE, warning=FALSE}
# Figure 3

metaT_metadata = readRDS('rds_files/metaT_metadata')

genome_metaT_files = system('ls ~/Documents/guaymas_analysis/flat_files/reads_mapped_per_genome/*', intern = TRUE)

read_genome_metaT_data = function(.coverM_genome_TPM_file) {
  read_tsv(.coverM_genome_TPM_file, show_col_types = FALSE) |>
    rename(bin = Genome) |>
    filter(if_any(.cols = 2, ~ !is.na(.x))) |> #filter by col number, not col name
    select(1, 3) |> # select bin name col & col containing % total metaT reads mapped per MAG
    rename_with(~ str_replace(.x, '_R1_paired_fastx.fastq Relative Abundance \\(%\\)', ''))
}

mag_data_cols = c('mag_name', 'phylum', 'class', 'order', 'completeness')

genome_metaT_data = genome_metaT_files |>
  map(function(.file) {read_genome_metaT_data(.file)}) |>
  reduce(left_join, by = 'bin') |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, bin, phylum, class, order, completeness),
            by = 'bin') |>
  select(-bin) |>
  relocate(all_of(mag_data_cols)) |>
  filter(!is.na(mag_name)) |>
  arrange(phylum) |>
  mutate(phylum = case_when(
    phylum == 'TA06_A' ~ 'Candidatus Rosutiota',
    phylum == '4484-113' ~ 'Candidatus Niteriota',
    TRUE ~ phylum)) |>
  mutate(`Phylum/Class` = if_else(class == 'Alphaproteobacteria', class, phylum)) |>
  relocate(`Phylum/Class`, .after = 1) |>
  group_by(`Phylum/Class`) |>
  add_tally(n(), name = 'total') |>
  ungroup() |>
  relocate(total) |>
  arrange(desc(total), phylum, order) |>
  select(-total)
#select(-c(class, order, completeness))

genome_metaT_rowAnno = genome_metaT_data |>
  select(mag_name, `Phylum/Class`) |>
  mutate(`Phylum/Class` = as_factor(`Phylum/Class`)) |>
  column_to_rownames('mag_name')

genome_metaT_colAnno = metaT_metadata |>
  column_to_rownames('sample') |>
  rename(
    `Depth (mbsf)` = depth,
    Site = site) |>
  select(-Site)


metaT_env = read_tsv('~/Downloads/david_geochemistry_1.txt') |> 
  rename(sample = Sample,
         Temperature = 6) |>
  mutate(sample = sample |> str_replace_all(
    c('U' = '', '-' = '')))
  

genome_metaT_colAnno = metaT_metadata |>
  rename(
    `Depth (mbsf)` = depth,
    Site = site) |>
  select(-Site) |> 
  mutate(sample = sample |> str_replace('_', 'B')) |> 
  left_join(metaT_env |> 
              select(sample, Temperature),
            by = 'sample') |> 
  rename(Temperature = 3) |>
  mutate(sample = sample |> str_replace('(.*B)(.*)', '\\1_\\2'),
         sample = paste0('U', sample)) |> 
  column_to_rownames('sample')


genome_metaT_heat_data = genome_metaT_data |>
  select(-c(`Phylum/Class`, phylum, class, order, completeness)) |>
  column_to_rownames('mag_name')


metaT_mag_row_order = guay_row_anno |>
  rownames()

genome_metaT_colAnno_onlyMetagenomeSites = genome_metaT_colAnno |>
  rownames_to_column('sample') |>
  filter(str_detect(sample, '1545|1546|1547|1548|1549')) |>
  column_to_rownames('sample')

genome_metaT_heat_onlyMetagenomeSites = genome_metaT_heat_data |>
  relocate(metaT_metadata$sample) |>
  select(matches('1545.*|1546.*|1547.*|1548.*|1549.*'))


genome_metaT_rowAnno2 = genome_metaT_rowAnno |>
  rownames_to_column('mag_name') |>
  arrange(factor(mag_name, levels = metaT_mag_row_order)) |>
  column_to_rownames('mag_name') |> 
  mutate(`Phylum/Class` = fct_inorder(`Phylum/Class`))



heat_figure2_metaT = metaT_env |> 
  select(sample, `Depth (mbsf)`, Temperature) |> 
  rename(col = sample, depth = `Depth (mbsf)`, temperature = Temperature) |> 
  mutate(depth = round(depth, digits = 1),
    new_colname = col |> 
           str_replace('(.*)(\\d+H\\d+)', paste0('U\\1_', depth))) |> 
  arrange(temperature) |> 
  filter(str_detect(col, '1545.*|1546.*|1547.*|1548.*|1549.*'))


metaT_heatmap =
  genome_metaT_heat_onlyMetagenomeSites |>
  rownames_to_column('mag_name') |>
  as_tibble() |>
  arrange(factor(mag_name, levels = metaT_mag_row_order)) |>
  column_to_rownames('mag_name') |>
  rename_with(~ paste0('U', .x) |> str_replace('(.*)(_.*)', '\\1B\\2')) |> 
  relocate(genome_metaT_colAnno_onlyMetagenomeSites |> 
             arrange(Temperature) |> 
             rownames()) |> 
  rename_with(~ heat_figure2_metaT |> 
                arrange(temperature) |> 
                pull(new_colname)) |> 
  as.matrix() |> 
  
  ComplexHeatmap::pheatmap(
    name = 'Total reads mapped (%)',
    main = 'MAG metatranscriptomic read recruitment',
    gaps_row = 63,
    gaps_col = c(8,12),
    show_colnames = TRUE,
    show_rownames = FALSE,
    annotation_col = genome_metaT_colAnno_onlyMetagenomeSites |> 
      arrange(Temperature) |> 
      rename(`Temperature (°C)` = Temperature),
    annotation_row = genome_metaT_rowAnno2,
    annotation_colors = guayMagRelPalette,
    cluster_rows = FALSE,
    cluster_cols = FALSE,
    border_color = 'grey30',
    treeheight_row = 0,
    treeheight_col = 10,
    color = c('grey60', rev(viridis::magma(n = 50))),
    show_row_dend = FALSE,
    show_column_dend = TRUE
    )

ComplexHeatmap::draw(metaT_heatmap, merge_legend = TRUE)


# pdf(
#   file = '~/Documents/guaymas_analysis/may_25_all_manuscript_docs/figure_3.pdf',
#   width = 7,
#   height = 9)
# ComplexHeatmap::draw(metaT_heatmap,
#      merge_legend = TRUE)
# dev.off()

```

<br> 

## Figure 4. Non-metric multidimensional scaling (nMDS) ordination plot of metagenomes reconstructed from sites U1545B, U1546B, U1547B, U1548B, U1549B and environmental parameters with a significant effect on microbial community composition (by temperature).
```{r echo=TRUE, message=FALSE, warning=FALSE}
env_metadata3 = readRDS('rds_files/env_metadata3.rds')

guay_tpm_files = system('ls flat_files/mag_relative_abundance_coverM/new_tpm_abun_dec_2022/*.tsv', intern = TRUE)

guay_tpm_tbl = guay_tpm_files |>
  map(~ read_tsv(.x, show_col_types = FALSE)) |>
  reduce(left_join, by = 'Genome') |>
  rename_with(~ str_replace(.x, '(.*)_(.*)_S\\d+.*', '\\1\\2')) |>
  rename_with(~ str_replace(.x, '(.*)-(.*)_S\\d+.*', '\\1_\\2')) |>
  rename_with(~ str_replace(.x, '^U(.*B)(.*)', '\\1_\\2')) |>
  rename(bin = Genome) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, bin), by = 'bin') |>
  relocate(mag_name, .before = 1) |>
  filter(!is.na(mag_name)) |>
  select(-bin)

guay_tpm_pcoa_data = guay_tpm_tbl |>
  pivot_longer(
    cols = 2:last_col(),
    names_to = 'sample',
    values_to = 'value') |>
  left_join(metagenome_sample_metadata |>
              rename(sample = Sample),
            by = 'sample') |>
  select(-`Depth (mbsf)`) |>
  pivot_wider(
    names_from = sample,
    values_from = value) |>
  unnest(everything()) |>
  relocate(c(mag_name, metagenome_sample_metadata$Sample)) |>
  pivot_longer(
    cols = 2:last_col(),
    names_to = 'sample',
    values_to = 'relative_abundance') |>
  pivot_wider(
    names_from = mag_name,
    values_from = relative_abundance) |>
  column_to_rownames('sample')

guay_tpm_pcoa_data_FINAL = guay_tpm_pcoa_data

set.seed(123)
dist = guay_tpm_pcoa_data_FINAL |>
  vegdist(method = 'bray')


nmds_result_FINAL = metaMDS(
  dist,
  trymax = 10000,
  autotransform = TRUE)


nmds_scores_FINAL = scores(nmds_result_FINAL, "site") |>
  as.data.frame() %>%
  mutate(sample = rownames(.),
         sample = paste0('U', sample)) |>
  as_tibble() |>
  left_join(env_metadata3, by = 'sample') |>
  mutate(Site = str_replace(
    sample, '(\\d{4}[:upper:]).*', '\\1'))


FINAL_nmds_envFit = envfit(
  nmds_result_FINAL,
  env_metadata3[-c(1,2)],
  na.rm = TRUE,
  permutations = 10000)


FINAL_nmds_envScores = as.data.frame(scores(FINAL_nmds_envFit, display = 'vectors')) %>%
  mutate(env_variable = rownames(.),
         pvalue = FINAL_nmds_envFit$vectors$pvals) |>
  filter(pvalue <= 0.05)

FINAL_nmds_envScores


nmds_scores_FINAL = nmds_scores_FINAL |>
  rename(depth = `Depth (mbsf)`) |>
  mutate(depth_bin = case_when(
    depth < 15 ~ 'Shallow (0-15 mbsf)',
    depth >= 15 & depth < 60 ~ 'Intermediate (15-60 mbsf)',
    depth >= 60 ~ 'Deep (> 60 mbsf)'),
    
    temp_bin = case_when(
    `T` > 2 & `T` <= 20 ~ 'Cool (2-20°C)',
    `T` > 20 & `T` <= 45 ~ 'Temperate (20-45°C)',
    `T` > 45 ~ 'Hot (45-60°C)'), .after = `T`)


en_coord_cont = as.data.frame(scores(FINAL_nmds_envFit, "vectors")) *
  ordiArrowMul(FINAL_nmds_envFit)

en_coord_cont = en_coord_cont |>
  rownames_to_column('parameter') |>
  filter(parameter %in% FINAL_nmds_envScores$env_variable) |>
  column_to_rownames('parameter')



# figure 4 - nmds by temperature
mag_nmds_may_2023 = nmds_scores_FINAL |>
  mutate(temp_bin = if_else(temp_bin == 'Temperate (20-45°C)', 'Warm (20-45°C)', temp_bin)) |>

  ggplot(aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(fill = Site, shape = factor(
    temp_bin,
    levels = c('Cool (2-20°C)',
               'Warm (20-45°C)',
               'Hot (45-60°C)'))),
    size = 3) +
  geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2),
               data = en_coord_cont, size = 0.65, alpha = 0.5, colour = "grey60",
               arrow = arrow(length = unit(0.25, 'cm'))) +
  geom_text(data = en_coord_cont, aes(x = NMDS1, y = NMDS2), colour = "grey30",
            fontface = "bold", label = row.names(en_coord_cont)) +
  theme(axis.title = element_text(size = 14, face = "bold", colour = "grey30"),
        panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "grey30"),
        legend.title = element_text(size = 11, face = "bold", colour = "grey30"),
        legend.text = element_text(size = 10, colour = "grey30"),
        legend.position = 'top',
        legend.box = 'vertical',
        legend.key = element_blank(),
        legend.margin = margin(b = -5),) +
  labs(x = '\nNMDS1',
       y = 'NMDS2\n',
       shape = 'Temperature regime',
       fill = 'Sampling site') +
  scale_fill_manual(values = c(
    'U1545B' = 'mediumblue',
    'U1546B' = 'darkslategray1',
    'U1547B' = 'darkorchid4',
    'U1548B' = 'goldenrod1',
    'U1549B' = 'darkolivegreen4')) +
  scale_shape_manual(values = c(
    'Cool (2-20°C)' = 21,
    'Warm (20-45°C)' = 24,
    'Hot (45-60°C)' = 22)) +
  scale_color_manual(values = 'black') +
  guides(fill = guide_legend(override.aes = list(shape = 21)))
mag_nmds_may_2023


# ggsave(
#   filename = '~/Documents/guaymas_analysis/may_25_all_manuscript_docs/figure_4.pdf',
#   plot = mag_nmds_may_2023,
#   width = 8,
#   height = 8
# )


```
<br>

## Figure 5. Genome size analysis (boxplots and linear regression)

```{r echo=TRUE, message=FALSE, warning=FALSE}
library(tidyverse)
library(vroom)
library(ComplexHeatmap)
library(gridtext)
library(RColorBrewer)
library(cowplot)
library(latex2exp)

guay_relAbun_files = system('ls flat_files/mag_relative_abundance_coverM/new_rel_abun_dec_2022/*.tsv', intern = TRUE)

abund = guay_relAbun_files |>
  map(~ read_tsv(.x, show_col_types = FALSE)) |>
  reduce(left_join, by = 'Genome') |>
  rename_with(~ str_replace(.x, '(.*)_(.*)_S\\d+.*', '\\1\\2')) |> #colnames()
  rename_with(~ str_replace(.x, '(.*)-(.*)_S\\d+.*', '\\1_\\2')) |>  #colnames()
  rename_with(~ str_replace(.x, '^U(.*B)(.*)', '\\1_\\2')) |>
  rename_with(~ str_replace(.x, '(.*)', 'U\\1'), .cols = 2:last_col()) |>
  rename(bin = Genome) |>
  #slice(-1) |>  # remove the unmapped % information; won't be used in the heatmap
  mutate(across(2:last_col(), ~ round(.x, digits = 1))) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |> select(mag_name, bin), by = 'bin') |>
  relocate(mag_name, .before = 1) |>
  filter(!is.na(mag_name), bin != 'unmapped') |>
  select(-bin)




# significance test - by depth

sample_groups_depth = env_metadata3 |>
  rename(depth = 2, temperature = 3) |>
  select(sample, depth, temperature) |>
  mutate(depth_group = case_when(
    depth <= 15 ~ 'shallow',
    depth > 15 & depth <= 60 ~ 'intermediate',
    depth > 60 ~ 'deep'),
    temperature_group = case_when(
      temperature <= 20 ~ 'cool',
      temperature > 20 & temperature <= 45 ~ 'warm',
      temperature > 45 ~ 'hot')) |>
  group_by(depth_group) |>
  group_split() %>%
  set_names(map_chr(., ~ unique(.x$depth_group)))

shallow_samples = sample_groups_depth$shallow
intermediate_samples = sample_groups_depth$intermediate
deep_samples = sample_groups_depth$deep


shallow_mags = abund |>
  select(mag_name, shallow_samples$sample) |>
  filter(if_any(.cols = 2:last_col(), ~ .x > 0)) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              mutate(completeness = completeness / 100,
                est_genome_size = size / completeness) |>
              select(mag_name, est_genome_size),
            by = 'mag_name') |>
  relocate(est_genome_size, .after = mag_name)

intermediate_mags = abund |>
  select(mag_name, intermediate_samples$sample) |>
  filter(if_any(.cols = 2:last_col(), ~ .x > 0)) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              mutate(completeness = completeness / 100,
                     est_genome_size = size / completeness) |>
              select(mag_name, est_genome_size),
            by = 'mag_name') |>
  relocate(est_genome_size, .after = mag_name)


deep_mags = abund |>
  select(mag_name, deep_samples$sample) |>
  filter(if_any(.cols = 2:last_col(), ~ .x > 0)) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              mutate(completeness = completeness / 100,
                     est_genome_size = size / completeness) |>
              select(mag_name, est_genome_size),
            by = 'mag_name') |>
  relocate(est_genome_size, .after = mag_name)


# mean(shallow_mags$est_genome_size)
# mean(intermediate_mags$est_genome_size)
# mean(deep_mags$est_genome_size)
# mean(shallow_mags$est_genome_size) / mean(deep_mags$est_genome_size)


# two-sided Welch's t-test
t.test(shallow_mags$est_genome_size, deep_mags$est_genome_size)
t.test(intermediate_mags$est_genome_size, deep_mags$est_genome_size)
t.test(shallow_mags$est_genome_size, intermediate_mags$est_genome_size)

# adjust the above p-values for multiple comparisons
p.adjust(c(0.02711, 0.05877, 0.616), method = 'BH')


mags_depth = shallow_mags |>
  select(mag_name, est_genome_size) |>
  mutate(depth_group = 'shallow') |>
  bind_rows(
    intermediate_mags |>
      select(mag_name, est_genome_size) |>
      mutate(depth_group = 'intermediate')) |>
  bind_rows(
    deep_mags |>
      select(mag_name, est_genome_size) |>
      mutate(depth_group = 'deep'))


# by temperature

sample_groups_temperature = env_metadata3 |>
  rename(depth = 2, temperature = 3) |>
  select(sample, depth, temperature) |>
  mutate(depth_group = case_when(
    depth <= 15 ~ 'shallow',
    depth > 15 & depth <= 60 ~ 'intermediate',
    depth > 60 ~ 'deep'),
    temperature_group = case_when(
      temperature <= 20 ~ 'cool',
      temperature > 20 & temperature <= 45 ~ 'warm',
      temperature > 45 ~ 'hot')) |>
  group_by(temperature_group) |>
  group_split() %>%
  set_names(map_chr(., ~ unique(.x$temperature_group)))

cool_samples = sample_groups_temperature$cool
warm_samples = sample_groups_temperature$warm
hot_samples = sample_groups_temperature$hot


cool_mags = abund |>
  select(mag_name, cool_samples$sample) |>
  filter(if_any(.cols = 2:last_col(), ~ .x > 0)) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              mutate(completeness = completeness / 100,
                     est_genome_size = size / completeness) |>
              select(mag_name, est_genome_size),
            by = 'mag_name') |>
  relocate(est_genome_size, .after = mag_name)

warm_mags = abund |>
  select(mag_name, warm_samples$sample) |>
  filter(if_any(.cols = 2:last_col(), ~ .x > 0)) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              mutate(completeness = completeness / 100,
                     est_genome_size = size / completeness) |>
              select(mag_name, est_genome_size),
            by = 'mag_name') |>
  relocate(est_genome_size, .after = mag_name)


hot_mags = abund |>
  select(mag_name, hot_samples$sample) |>
  filter(if_any(.cols = 2:last_col(), ~ .x > 0)) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              mutate(completeness = completeness / 100,
                     est_genome_size = size / completeness) |>
              select(mag_name, est_genome_size),
            by = 'mag_name') |>
  relocate(est_genome_size, .after = mag_name)


# mean(cool_mags$est_genome_size)
# mean(warm_mags$est_genome_size)
# mean(hot_mags$est_genome_size)
mean(cool_mags$est_genome_size) / mean(hot_mags$est_genome_size)
mean(warm_mags$est_genome_size) / mean(hot_mags$est_genome_size)

# two-sided Welch's t-test
t.test(cool_mags$est_genome_size, hot_mags$est_genome_size)
t.test(warm_mags$est_genome_size, hot_mags$est_genome_size)
t.test(cool_mags$est_genome_size, warm_mags$est_genome_size)

# adjust the above p-values for multiple comparisons
p.adjust(c(0.001406, 0.03694, 0.209), method = 'BH')



mags_temperature = cool_mags |>
  select(mag_name, est_genome_size) |>
  mutate(temperature_group = 'cool') |>
  bind_rows(
    warm_mags |>
      select(mag_name, est_genome_size) |>
      mutate(temperature_group = 'warm')) |>
  bind_rows(
    hot_mags |>
      select(mag_name, est_genome_size) |>
      mutate(temperature_group = 'hot'))



# boxplots

# temperature boxplot
temperature_mag_size_boxplot_plot = mags_temperature |>
  mutate(
    est_genome_size = est_genome_size / 1e06,
    temperature_group = case_when(
      temperature_group == 'cool' ~ 'Cool (2-20°C)',
      temperature_group == 'warm' ~ 'Warm (20-45°C)',
      temperature_group == 'hot' ~ 'Hot (>45°C)'),

    temperature_group = factor(temperature_group,
                               levels = c('Cool (2-20°C)',
                                          'Warm (20-45°C)',
                                          'Hot (>45°C)'))) |>
  ggplot(aes(
    y = est_genome_size,
    x = temperature_group,
    fill = temperature_group)) +
  geom_boxplot(
    outlier.shape = NA,
    width = 0.5) +
  stat_summary(fun = mean, geom = "point", shape = 23,
               size = 3, fill = "white", color = "black") +
  geom_jitter(color = "black", size = 0.7, alpha = 0.5) +
  theme_bw() +
  theme(
    axis.title.y = element_text(size = 12),
    axis.text.y = element_text(size = 11),
    panel.grid = element_blank(),
    axis.text.x = element_text(size = 11, angle = 10, hjust = 0.5, vjust = 0.5),
    axis.title.x = element_blank(),
    #axis.ticks.x = element_blank(),
    legend.position = 'none') +
  labs(y = 'Estimated total genome size (Mbp)\n',
       fill = 'Temperature regime') +
  scale_fill_manual(
    values = c(
      'Cool (2-20°C)' = 'grey95',
      'Warm (20-45°C)' = 'grey70',
      'Hot (>45°C)' = 'grey45'
    )
  ) +
  #p-values
  annotate('text', x = 1.5, y = 7.5,
           label = 'italic(p) == 0.209',
           size = 3.4, parse = TRUE) +
  annotate('text', x = 2, y = 6.75,
           label = 'italic(p) == 0.004',
           size = 3.4, parse = TRUE) +
  annotate('text', x = 2.5, y = 6,
           label = 'italic(p) == 0.055',
           size = 3.4, parse = TRUE) +

  #segments
  annotate('segment', x = 1, xend = 2,
           y = 7.25, yend = 7.25,
           colour = 'black', size = 0.3) +
  annotate('segment', x = 1, xend = 3,
           y = 6.5, yend = 6.5,
           colour = 'black', size = 0.3) +
  annotate('segment', x = 2, xend = 3,
           y = 5.75, yend = 5.75,
           colour = 'black', size = 0.3) +

  scale_y_continuous(breaks = seq(0.5, 6, by = 1),
                     limits = c(0.5, 7.5))

temperature_mag_size_boxplot_plot





# depth boxplot
depth_mag_size_boxplot_plot = mags_depth |>
  mutate(
    est_genome_size = est_genome_size / 1e06,
    depth_group = case_when(
      depth_group == 'shallow' ~ 'Shallow (2-15 mbsf)',
      depth_group == 'intermediate' ~ 'Intermediate (15-60 mbsf)',
      depth_group == 'deep' ~ 'Deep (>60 mbsf)'),

    depth_group = factor(depth_group,
                         levels = c('Shallow (2-15 mbsf)',
                                    'Intermediate (15-60 mbsf)',
                                    'Deep (>60 mbsf)'))) |>
  ggplot(aes(
    y = est_genome_size,
    x = depth_group,
    fill = depth_group)) +
  geom_boxplot(
    outlier.shape = NA,
    width = 0.5) +
  stat_summary(fun = mean, geom = "point",
               shape = 23, size = 3, fill = "white", color = "black") +
  geom_jitter(color = "black", size = 0.7, alpha = 0.5) +
  theme_bw() +
  theme(
    axis.text.y = element_text(size = 11),
    panel.grid = element_blank(),
    axis.text.x = element_text(size = 11, angle = 10, hjust = 0.5, vjust = 0.5),
    axis.title.x = element_blank(),
    legend.position = 'none',
    ) +
  labs(y = '\n',
       fill = 'Depth regime') +
  scale_fill_manual(values = c(
    'Shallow (2-15 mbsf)' = 'grey95',
    'Intermediate (15-60 mbsf)' = 'grey70',
    'Deep (>60 mbsf)' = 'grey45'
  )) +
  #p-values
  annotate('text', x = 1.5, y = 7.5,
           label = 'italic(p) == 0.62',
           size = 3.4, parse = TRUE) +
  annotate('text', x = 2, y = 6.75,
           label = 'italic(p) == 0.081',
           size = 3.4, parse = TRUE) +
  annotate('text', x = 2.5, y = 6,
           label = 'italic(p) == 0.088',
           size = 3.4, parse = TRUE) +

  #segments
  annotate('segment', x = 1, xend = 2,
           y = 7.25, yend = 7.25,
           colour = 'black', size = 0.3) +
  annotate('segment', x = 1, xend = 3,
           y = 6.5, yend = 6.5,
           colour = 'black', size = 0.3) +
  annotate('segment', x = 2, xend = 3,
           y = 5.75, yend = 5.75,
           colour = 'black', size = 0.3) +

  scale_y_continuous(breaks = seq(0.5, 6, by = 1),
                     limits = c(0.5, 7.5))

depth_mag_size_boxplot_plot


# ggsave(
#   plot = depth_temperature_genome_size_boxplots,
#   filename = '~/Documents/guaymas_analysis/final_figures_and_tables/figure_5_boxplots.pdf',
#   width = 14,
#   height = 6,
#   dpi = 500,
#   units = 'in')






# linear regression


# temperature regression
mags_temperature_regression =
  cool_mags |>
  pivot_longer(cols = 3:last_col(),
               names_to = 'sample',
               values_to = 'percent_reads_mapped') |>
  filter(percent_reads_mapped > 0) |>
  mutate(temperature_group = 'cool') |>
  left_join(env_metadata3 |>
              select(sample, `T`),
            by = 'sample') |>
  #pull(`T`) |> is.na() |> table()
  bind_rows(
    warm_mags |>
      pivot_longer(cols = 3:last_col(),
                   names_to = 'sample',
                   values_to = 'percent_reads_mapped') |>
      filter(percent_reads_mapped > 0) |>
      mutate(temperature_group = 'cool') |>
      left_join(env_metadata3 |>
                  select(sample, `T`),
                by = 'sample') |>
      mutate(temperature_group = 'temperate')
  ) |>
  bind_rows(
    hot_mags |>
      pivot_longer(cols = 3:last_col(),
                   names_to = 'sample',
                   values_to = 'percent_reads_mapped') |>
      filter(percent_reads_mapped > 0) |>
      mutate(temperature_group = 'cool') |>
      left_join(env_metadata3 |>
                  select(sample, `T`),
                by = 'sample') |>
      mutate(temperature_group = 'hot')
  ) |>
  rename(temperature = `T`)



# run regression
temperature_regression_data = mags_temperature_regression |>
  mutate(
    est_genome_size = est_genome_size / 1e06) |>
  group_by(sample, temperature) |>
  summarize(ags = mean(est_genome_size), .groups = 'drop')

temperature_regression = lm(ags ~ temperature, data = temperature_regression_data)
summary(temperature_regression)


temperature_mag_size_regression_plot =
  mags_temperature_regression |>
  mutate(
    est_genome_size = est_genome_size / 1e06,
    temperature_group = case_when(
      temperature_group == 'cool' ~ 'Cool (2-20°C)',
      temperature_group == 'temperate' ~ 'Temperate (20-45°C)',
      temperature_group == 'hot' ~ 'Hot (>45°C)'),
    temperature_group = factor(temperature_group,
                               levels = c('Cool (2-20°C)',
                                          'Temperate (20-45°C)',
                                          'Hot (>45°C)'))
  ) |>
  group_by(sample, temperature, temperature_group) |>
  summarize(ags = mean(est_genome_size), .groups = 'drop') |>
  ggplot(aes(
    y = ags,
    x = temperature
  )) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x) +
  annotate('text', x = 45, y = 2.4,
           label = 'italic(p) == 7.39~x~10^{-7}',
           size = 4.2, parse = TRUE) +
  annotate('text', x = 42.7, y = 2.25,
           label = TeX('$R^{2}_{adj} = .65'),
           size = 4.2, parse = TRUE) +
  theme_bw() +
  theme(
    axis.title = element_text(size = 12),
    axis.text = element_text(size = 11),
    panel.grid = element_blank(),
    legend.position = 'top') +
  labs(y = 'Average genome size (Mbp)\n',
       x = 'Temperature (°C)') +
  scale_y_continuous(breaks = seq(1, 2.5, by = 0.5),
                     limits = c(1.0, 2.5))

temperature_mag_size_regression_plot



# depth regression

mags_depth_regression =
  shallow_mags |>
  pivot_longer(cols = 3:last_col(),
               names_to = 'sample',
               values_to = 'percent_reads_mapped') |>
  filter(percent_reads_mapped > 0) |>
  mutate(temperature_group = 'shallow') |>
  left_join(env_metadata3 |>
              select(sample, `Depth (mbsf)`),
            by = 'sample') |>
  #pull(`Depth (mbsf)`) |> is.na() |> table()
  bind_rows(
    intermediate_mags |>
      pivot_longer(cols = 3:last_col(),
                   names_to = 'sample',
                   values_to = 'percent_reads_mapped') |>
      filter(percent_reads_mapped > 0) |>
      mutate(depth_group = 'intermediate') |>
      left_join(env_metadata3 |>
                  select(sample, `Depth (mbsf)`),
                by = 'sample')
  ) |>
  bind_rows(
    deep_mags |>
      pivot_longer(cols = 3:last_col(),
                   names_to = 'sample',
                   values_to = 'percent_reads_mapped') |>
      filter(percent_reads_mapped > 0) |>
      mutate(temperature_group = 'deep') |>
      left_join(env_metadata3 |>
                  select(sample, `Depth (mbsf)`),
                by = 'sample') |>
      mutate(temperature_group = 'deep')
  ) |>
  rename(depth = `Depth (mbsf)`)



# run regression
depth_regression_data = mags_depth_regression |>
  mutate(
    est_genome_size = est_genome_size / 1e06) |>
  group_by(sample, depth) |>
  summarize(ags = mean(est_genome_size), .groups = 'drop')

depth_regression = lm(ags ~ depth, data = depth_regression_data)
summary(depth_regression)



# plot results
depth_mag_size_regression_plot =
  mags_depth_regression |>
  mutate(
    est_genome_size = est_genome_size / 1e06) |>
  group_by(sample, depth) |>
  summarize(ags = mean(est_genome_size), .groups = 'drop') |>
  ggplot(aes(
    y = ags,
    x = depth
  )) +
  geom_point() +
  geom_smooth(method = 'lm', formula= y ~ x) +
  annotate('text', x = 172, y = 2.4,
           label = 'italic(p) == 1.36~x~10^{-6}',
           size = 4.2, parse = TRUE) +
  annotate('text', x = 162, y = 2.25,
           label = TeX('$R^{2}_{adj} = .63'),
           size = 4.2, parse = TRUE) +
  theme_bw() +
  theme(
    axis.title = element_text(size = 12),
    axis.text = element_text(size = 11),
    panel.grid = element_blank(),
    axis.title.y = element_blank(),
    legend.position = 'top') +
  labs(y = 'Average genome size (Mbp)\n',
       x = 'Depth (mbsf)') +
  scale_y_continuous(breaks = seq(1, 2.5, by = 0.5),
                     limits = c(1.0, 2.5))


depth_mag_size_regression_plot

figure_5_genome_size =
  plot_grid(
    ncol = 4,
    nrow = 2,
    rel_widths = c(0.11, 1, 0.09, 1),
    labels = c('A','','  B','','C','','  D',''),
    NULL,
    temperature_mag_size_boxplot_plot,
    NULL,
    depth_mag_size_boxplot_plot,

    NULL,
    temperature_mag_size_regression_plot,
    NULL,
    depth_mag_size_regression_plot,
    align = 'hv'
    )

figure_5_genome_size



# ggsave(
#   plot = figure_5_genome_size,
#   filename = '~/Documents/guaymas_analysis/may_25_all_manuscript_docs/figure_5.pdf',
#   width = 10,
#   height = 10,
#   dpi = 500,
#   units = 'in')

```

## Figure 6. MAG recovery at sampled sites vs. temperature and depth. 

```{r echo=TRUE, message=FALSE, warning=FALSE}
# figure 6

step_fn_data = guay_relAbun_tbl |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, domain, phylum)) |>
  relocate(c(domain, phylum), .after = 1) |>
  pivot_longer(
    4:last_col(),
    names_to = 'sample',
    values_to = 'abundance') |>
  mutate(abundance = round(abundance, digits = 2),
         sample = paste0('U', sample)) |>
  left_join(guay_col_anno |>
              rownames_to_column('sample') |>
              rename(depth = `Depth (mbsf)`,
                     temperature = `T`) |>
              select(sample, depth, temperature)) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              mutate(completeness = completeness / 100,
                     est_total_genome_size = size / completeness) |>
              rename(genome_size = size) |> 
              select(mag_name, est_total_genome_size, genome_size),
            by = 'mag_name') |> 
  # left_join(andreas_mag_size_plot_tbl2 |>
  #             select(mag_name, est_total_genome_size, genome_size)) |>
  mutate(depth_group = case_when(
    depth <= 15 ~ 'shallow',
    depth > 15 & depth <= 60 ~ 'intermediate',
    depth > 60 ~ 'deep'),
    temperature_group = case_when(
      temperature <= 20 ~ 'psychrophilic',
      temperature > 20 & temperature <= 45 ~ 'mesophilic',
      temperature > 45 ~ 'thermophilic'))


step_fn_data = step_fn_data |> 
  filter(abundance > 0) |> 
  group_by(sample, depth, temperature, 
           depth_group, temperature_group) |> 
  summarize(n = n(), .groups = 'drop') |> 
  mutate(group = if_else(
    str_detect(sample, '1547|1548'),
    'Ringvent', 'Other'),
    group = factor(group, levels = c('Ringvent', 'Other')))


#temperature

# fit regression, extract p-value and adj. R^2
nmag_temp_regression_ringvent = lm(n ~ temperature, 
                                   data = (step_fn_data |> 
                                             filter(group == 'Ringvent')))
summary(nmag_temp_regression_ringvent)

nmag_temp_regression_other = lm(n ~ temperature, 
                                data = (step_fn_data |> 
                                          filter(group == 'Other')))
summary(nmag_temp_regression_other)



nmag_temp_reg = step_fn_data |> 
  ggplot(aes(x = temperature, y = n, color = group)) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x) +
  theme_bw() + 
  theme(panel.grid = element_blank(),
        legend.position = c(0.865, 0.8875),
        legend.background = element_rect(fill = "white", color = "black")) +
  labs(
    y = 'Number of unique MAGs\n',
    x = '\nTemperature (°C)',
    color = 'Site') +
  annotate('text', x = 44.25, y = 55,
           label = 'italic(p) == 1.11~x~10^{-6}~(Ringvent)',
           size = 3, parse = TRUE) +
  annotate('text', x = 42.5, y = 48,
           label = TeX('$R^{2}_{adj} = .86~(Ringvent)'),
           size = 3, parse = TRUE) +
  
  annotate('text', x = 14.25, y = 15,
           label = 'italic(p) == 2.96~x~10^{-5}~(Other)',
           size = 3, parse = TRUE) +
  annotate('text', x = 12.5, y = 8,
           label = TeX('$R^{2}_{adj} = .82~(Other)'),
           size = 3, parse = TRUE)



# depth

# fit linear regression

nmag_depth_regression_ringvent = lm(n ~ depth, 
                                   data = (step_fn_data |> 
                                             filter(group == 'Ringvent')))
summary(nmag_depth_regression_ringvent)

nmag_depth_regression_other = lm(n ~ depth, 
                                data = (step_fn_data |> 
                                          filter(group == 'Other')))
summary(nmag_depth_regression_other)


#plot
nmag_depth_reg = step_fn_data |> 
  ggplot(aes(x = depth, y = n, color = group)) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x) +
  theme_bw() + 
  theme(panel.grid = element_blank(),
        axis.title.y = element_blank(),
        legend.position = c(0.865, 0.8875),
        legend.background = element_rect(fill = "white", color = "black")) +
  labs(
    #y = 'Number of unique MAGs\n',
    x = '\nDepth (mbsf)',
    color = 'Site') +
  annotate('text', x = 178, y = 55,
           label = 'italic(p) == 2.70~x~10^{-5}~(Other)',
           size = 3, parse = TRUE) +
  annotate('text', x = 172.25, y = 48,
           label = TeX('$R^{2}_{adj} = .83~(Other)'),
           size = 3, parse = TRUE) +
  annotate('text', x = 124, y = -4,
           label = 'italic(p) == 1.77~x~10^{-7}~(Ringvent)',
           size = 3, parse = TRUE) +
  annotate('text', x = 115, y = -12,
           label = TeX('$R^{2}_{adj} = .90~(Ringvent)'),
           size = 3, parse = TRUE)


figure_6 =
  plot_grid(
    ncol = 4,
    nrow = 1,
    rel_widths = c(0.11, 1, 0.11, 1),
    labels = c('A', '', '  B', ''),
    NULL, nmag_temp_reg, NULL, nmag_depth_reg
    
    #align = 'hv'
  )

figure_6



# ggsave(
#   plot = figure_6,
#   filename = '~/Documents/guaymas_analysis/may_25_all_manuscript_docs/figure_6.pdf',
#   width = 10,
#   height = 5,
#   dpi = 500,
#   units = 'in')


```

<br>

## Supplementary Figure 2. MAG recovery frequency. Frequency of Guaymas Basin prokaryotic MAGs (≥ 50% completeness, ≤ 10% contamination) by bacterial (A) and archaeal phylum (B). 
```{r echo=TRUE, message=FALSE, warning=FALSE}
heat_colors <- c(colorRampPalette(colors = 'white')(1),
                 colorRampPalette(colors = c('#ABD9E9',
                                             '#FEE090',
                                             'orange',
                                             '#D73027'))(8000))



mag_dist_plot_archaea <- final_filtered_andreas_mag_summary_dec_2022 |>
  group_by(phylum) |>
  summarize(freq = n()) |>
  arrange(desc(freq)) |>
  mutate(x = row_number(),
         phylum) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(phylum, domain, completeness) |>
              group_by(phylum) |>
              add_tally(mean(completeness), name = 'average_completeness') |>
              select(-completeness) |>
              ungroup() |>
              distinct(),
            by = 'phylum') |>
  #mutate(phylum = if_else(is.na(phylum), 'Unclassified', phylum)) |>
  mutate(phylum = as_factor(phylum)) |>
  filter(domain == 'Archaea') |>
  ggplot(aes(phylum, freq)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle = 65, hjust = 1, size = 8),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_blank(),
        strip.text = element_text(size = 10),
        strip.background = element_rect(fill = 'wheat1'),
        axis.text.y = element_text(size = 8),
        legend.position = 'none'
  ) +
  geom_segment(aes(xend = phylum,  yend = 0), color = 'grey50') +
  geom_point(size = 1.5,
             aes(color = average_completeness)) +
  geom_text(aes(label = freq, vjust = -0.65), size = 2.5) +
  facet_grid(~ domain, scales = 'free', space = 'free') +
  labs(x = '\nPhylum',
       y = 'Number of MAGs per phylum\n',
       color = 'Average genome
completeness per phylum',

  ) +
  scale_color_gradientn(colors = heat_colors,
                        breaks = seq(60, 100, by = 5),
                        limits = c(64, 100)
  ) +
  scale_y_continuous(limits = c(0, 27),
                     expand = expansion(add = c(1, 0)))


mag_dist_plot_bacteria =
  final_filtered_andreas_mag_summary_dec_2022 |>
  #filter(!mag_name %in% c('MAG_1507', 'MAG_769', 'MAG_983')) |>
  group_by(phylum) |>
  summarize(freq = n()) |>
  arrange(desc(freq)) |>
  mutate(x = row_number(),
         phylum) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(phylum, domain, completeness)  |>
              group_by(phylum) |>
              add_tally(mean(completeness), name = 'average_completeness') |>
              select(-completeness) |>
              ungroup() |>
              distinct(),
            by = 'phylum') |>
  mutate(phylum = case_when(
    phylum == '4484-113' ~ 'Candidatus Niteriota',
    phylum == 'TA06_A' ~ 'Candidatus Rosutiota',
    is.na(phylum) ~ 'Unclassified',
    TRUE ~ phylum)) |>
  mutate(phylum = as_factor(phylum)) |>
  filter(domain == 'Bacteria') |>
  ggplot(aes(phylum, freq)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle = 65, hjust = 1, size = 8),
        axis.title = element_text(size = 10),
        strip.text = element_text(size = 10),
        strip.background = element_rect(fill = 'wheat1'),
        legend.title = element_text(size = 6),
        legend.direction = 'horizontal',
        legend.margin = margin(c(4,15,4,4)),
        legend.text = element_text(size = 6),
        axis.text.y = element_text(size = 8),
        legend.position = c(0.565, 0.9),
        legend.background = element_rect(fill = "white",
                                         color = "black")
  ) +
  geom_segment(aes(xend = phylum,  yend = 0), color = 'grey50') +
  geom_point(size = 1.5,
             aes(color = average_completeness)) +
  geom_text(aes(label = freq, vjust = -0.65), size = 2.5) +
  facet_grid(~ domain, scales = 'free', space = 'free') +
  labs(x = '\nPhylum',
       y = 'Number of MAGs per phylum\n',
       color = 'Average genome
completeness per phylum',
  ) +
  scale_color_gradientn(colors = heat_colors,
                        breaks = seq(60, 100, by = 5),
                        limits = c(64, 100)
  ) +
  scale_y_continuous(limits = c(0, 27),
                     expand = expansion(add = c(1, 0)))


mag_dist_plot = plot_grid(NULL, mag_dist_plot_bacteria, NULL, mag_dist_plot_archaea,
                          labels = c('A', NA, 'B', NA),
                          rel_widths = c(0.25, 3, 0.2, 1.5),
                          nrow = 1,
                          label_size = 11.5,
                          align = 'hv',
                          label_x = c(0.045, 0.205))

mag_dist_plot

```




## Supplementary Figure 3. MAG relative abundances across taxonomic orders, modified from Figure 2. 
```{r echo=TRUE, message=FALSE, warning=FALSE}
# Supplementary Figure 2

guay_relAbun_complexHeatmap_order =
  guay_relAbun_plot_data |>
  left_join(
    final_filtered_andreas_mag_summary_dec_2022 |> 
      select(mag_name, order),
    by = 'mag_name') |> 
  mutate(mag_name = mag_name |> 
           str_replace('Guaymas_MAG_P_(\\d+)', paste0('GMP_\\1; ', order))) |> 
  select(-order) |> 
  column_to_rownames('mag_name') |>
  relocate(guay_col_anno |>
             select(`Depth (mbsf)`, `T`) |>
             rename(`Temperature (°C)` = `T`) |>
             arrange(`Temperature (°C)`) |>
             rownames()) |>
  rename_with(~ heat_figure2_new$col_names, .cols = everything()) |> 
  as.matrix() |>
  ComplexHeatmap::pheatmap(
    name = 'Relative abundance
(% total reads mapped)',
main = 'Guaymas MAG relative abundance',
show_colnames = TRUE,
annotation_row = guay_row_anno,
annotation_col = guay_col_anno |>
  select(`Depth (mbsf)`, `T`) |>
  rename(`Temperature (°C)` = `T`) |>
  arrange(`Temperature (°C)`),
annotation_colors = guayMagRelPalette,
cluster_rows = FALSE,
cluster_cols = FALSE,
show_rownames = TRUE,
gaps_row = 63,
gaps_col = c(13, 21),
border_color = 'grey30',
treeheight_row = 0,
treeheight_col = 10,
color = c('grey60', rev(viridis::magma(n = 50))),
legend_breaks = 0:3,
legend_labels = gt_render(c('0', sprintf('%.1f', round(expm1(1:3), 1)))),
show_row_dend = FALSE,
show_column_dend = TRUE)


draw(guay_relAbun_complexHeatmap_order,
     merge_legend = TRUE)


# pdf(
#   file = '~/Documents/guaymas_analysis/may_25_all_manuscript_docs/supplementary_figure_2.pdf',
#   width = 7,
#   height = 9)
# draw(guay_relAbun_complexHeatmap_order,
#      merge_legend = TRUE)
# dev.off()


```



## Supplementary Figure 3. Taxonomic order-specific relative abundance of Chloroflexota and Thermoproteota MAGs in metagenomic samples from drill sites U1545B and U1547B

```{r echo=TRUE, message=FALSE, warning=FALSE}
guay_relAbun_data_thermo_chloro_1545_1547 =
  guay_relAbun_plot_data |>
  #rename_with(~ paste0('U', .x), .cols = 2:last_col()) |> 
  relocate(mag_name,
           env_metadata3 |> 
             mutate(Site = sample |> str_replace('(.*)_.*', '\\1')) |> 
             arrange(Site, `T`) |> 
             pull(sample)
           ) |>
  pivot_longer(
    cols = 2:last_col(),
    names_to = 'sample',
    values_to = 'value') |>
  left_join(
    env_metadata3 |>
      rename(depth = 2) |> 
      select(sample, depth),
    by = 'sample') |>
  filter(str_detect(sample, '1545|1547'), depth <= 66) |>
  select(-depth) |>
  pivot_wider(
    names_from = sample,
    values_from = value) |>
  unnest(everything()) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, phylum, order),
            by = 'mag_name') |>
  filter(phylum %in% c('Thermoproteota', 'Chloroflexota')) |>
  arrange(phylum, order) |>
  column_to_rownames('mag_name') |>
  select(-c(phylum, order))


guay_relAbun_data_thermo_chloro_1545_1547_rowAnno =
  guay_relAbun_data_thermo_chloro_1545_1547 |>
  rownames_to_column('mag_name') |>
  select(mag_name) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              filter(phylum %in% c('Thermoproteota', 'Chloroflexota')) |>
              select(mag_name, phylum, class, order),
            by = 'mag_name') |>
  arrange(phylum, order) |>
  mutate(Order = order,
         Order = if_else(is.na(order), paste0(class, ' order'), Order),
         Order = as_factor(Order)) |>
  column_to_rownames('mag_name') |>
  select(-c(phylum, class, order))



guay_relAbun_data_thermo_chloro_1545_1547_colAnno =
   tibble(sample = colnames(guay_relAbun_data_thermo_chloro_1545_1547)) |> 
    left_join(env_metadata3 |> 
                select(1:3), #sample, depth, temp
              by = 'sample') |> 
    filter(str_detect(sample, '1545|1547'), `Depth (mbsf)` <= 66) |> 
    rename(`Temperature (°C)` = 3) |> 
    mutate(site = sample |> str_replace('U(.*)B_.*', '\\1')) |> 
    arrange(site, `Temperature (°C)`) |> 
    select(-site) |> 
    column_to_rownames('sample')



heat_thermo_chloro_new = tibble(
  col = colnames(guay_relAbun_data_thermo_chloro_1545_1547)) |> 
  left_join(env_metadata3 |> 
              select(1:3) |> 
              rename(col = sample))

heat_thermo_chloro_new = heat_thermo_chloro_new |> 
  select(-`T`) |> 
  rename(depth = 2) |> 
  mutate(col_names = col |> str_replace("(.*_).*", paste0("\\1", depth))) |> 
  select(-depth)



guay_relAbun_complexHeatmap_thermo_chloro_1545_1547 =
  guay_relAbun_data_thermo_chloro_1545_1547 |>
  rename_with(~ heat_thermo_chloro_new$col_names, .cols = everything()) |> 
  as.matrix() |>
    
  ComplexHeatmap::pheatmap(
    name = 'Relative abundance
(% total reads mapped)',
main = 'Guaymas Chloroflexota and Thermoproteota
relative abundance at sites U1545 and U1547',
show_colnames = TRUE,
annotation_row = guay_relAbun_data_thermo_chloro_1545_1547_rowAnno,
annotation_col = guay_relAbun_data_thermo_chloro_1545_1547_colAnno,
annotation_colors = list(
  `Depth (mbsf)` = c('white', 'black'),
  `Temperature (°C)` = c('white', 'firebrick3')),
gaps_col = c(5, 9, 13),
gaps_row = 23,
cluster_rows = FALSE,
cluster_cols = FALSE,
show_rownames = FALSE,
border_color = 'grey30',
treeheight_row = 0,
treeheight_col = 10,
color = c('grey60', rev(viridis::magma(n = 50))),
show_row_dend = FALSE,
show_column_dend = TRUE)


draw(guay_relAbun_complexHeatmap_thermo_chloro_1545_1547,
     merge_legend = TRUE)


# pdf(file = '~/Documents/guaymas_analysis/may_25_all_manuscript_docs/supplementary_figure_3.pdf',
#     width = 7,
#     height = 8)
# draw(guay_relAbun_complexHeatmap_thermo_chloro_1545_1547,
#      merge_legend = TRUE)
# dev.off()

```


<br>



## Supplementary Figure 4. Non-metric multidimensional scaling (nMDS) ordination plot of metagenomes reconstructed from sites U1545B, U1546B, U1547B, U1548B, U1549B and environmental parameters with a significant effect on microbial community composition (by depth).

```{r echo=TRUE, message=FALSE, warning=FALSE}
# supplementary figure 4; NMDS - by depth
mag_nmds_may_2023_depth = nmds_scores_FINAL |>
  mutate(depth_bin = if_else(depth_bin == 'Deep (> 60 mbsf)', 'Deep (>60 mbsf)', depth_bin)) |>

  ggplot(aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(shape = factor(
    depth_bin,
    levels = c('Shallow (0-15 mbsf)',
               'Intermediate (15-60 mbsf)',
               'Deep (>60 mbsf)')),
    fill = Site),
    size = 3) +
  #scale_colour_manual(values = c("orange", "steelblue"))  +
  geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2),
               data = en_coord_cont, size = 0.65, alpha = 0.5, colour = "grey60",
               arrow = arrow(length = unit(0.25, 'cm'))) +
  geom_text(data = en_coord_cont, aes(x = NMDS1, y = NMDS2), colour = "grey30",
            fontface = "bold", label = row.names(en_coord_cont)) +
  theme(axis.title = element_text(size = 14, face = "bold", colour = "grey30"),
        panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "grey30"),
        legend.title = element_text(size = 11, face = "bold", colour = "grey30"),
        legend.text = element_text(size = 10, colour = "grey30"),
        legend.position = 'top',
        legend.box = 'vertical',
        legend.key = element_blank(),
        legend.margin = margin(b = -5),) +
  labs(x = '\nNMDS1',
       y = 'NMDS2\n',
       shape = 'Depth regime',
       fill = 'Sampling site',) +
  scale_fill_manual(values = c(
    'U1545B' = 'mediumblue',
    'U1546B' = 'darkslategray1',
    'U1547B' = 'darkorchid4',
    'U1548B' = 'goldenrod1',
    'U1549B' = 'darkolivegreen4')) +
  scale_shape_manual(values = c(
    'Shallow (0-15 mbsf)' = 21,
    'Intermediate (15-60 mbsf)' = 24,
    'Deep (>60 mbsf)' = 22)) +
  scale_color_manual(values = 'black') +
  guides(fill = guide_legend(
    override.aes = list(shape = 21),
    order = 0),
    shape = guide_legend(order = 1))
mag_nmds_may_2023_depth

# 21 = circle
# 22 = square
# 24 = triangle

# ggsave(
#   filename = '~/Documents/guaymas_analysis/may_25_all_manuscript_docs/supplementary_figure_4.pdf',
#   plot = mag_nmds_may_2023_depth,
#   width = 8,
#   height = 8
# )

```




<br>

## Run Trimmomatic on metagenomes
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=compute # Queue selection
#SBATCH --job-name=imo # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/guaymas_andreas_metagenomes_trimmomatic_fastqc_nov_2022_%A-%a.log # Standard output/error
#SBATCH --array=0-12
#SBATCH --qos=unlim

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/Guaymas_Basin_Subsurface

module load anaconda/5.1
source activate trimmomatic

ADAPTER_FASTA=$(echo "/vortexfs1/home/dgellermcgrath/.conda/envs/trimmomatic/share/trimmomatic-0.39-2/adapters/NexteraPE-PE.fa")

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/Guaymas_Basin_Subsurface")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas")

FASTQC_OUTDIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/fastqc_of_trimmed_andreas_read_files")

FILE_PATTERN=("1545B_13H_4_S5" "1545B_19F_3_S6" "1545B_1H_2_S1" "1545B_2H_3_S2" "1545B_32F_3_S7" "1545B_4_H3_S3" "1545B_8_H3_S4" "1547B_1H_3_S8" "1547B_2H_3_S9" "1547B_3H_3_S10" "1547B_5H_3_S11" "1547B_7H_3_S12" "1547B_9H_3_S13")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


trimmomatic PE \
-threads 36 \
-trimlog ${OUT_DIR}/trim_logs/${FILE_PATTERN}.log \
-summary ${OUT_DIR}/trim_summaries/${FILE_PATTERN}_statsSummaryFile.txt \
${IN_DIR}/${FILE_PATTERN}_R1_001.fastq.gz \
${IN_DIR}/${FILE_PATTERN}_R2_001.fastq.gz \
${OUT_DIR}/paired_output/${FILE_PATTERN}_paired_1.fastq.gz \
${OUT_DIR}/unpaired_output/${FILE_PATTERN}_unpaired_1.fastq.gz \
${OUT_DIR}/paired_output/${FILE_PATTERN}_paired_2.fastq.gz \
${OUT_DIR}/unpaired_output/${FILE_PATTERN}_unpaired_2.fastq.gz \
ILLUMINACLIP:${ADAPTER_FASTA}:2:30:10 \
LEADING:20 \
TRAILING:20 \
SLIDINGWINDOW:04:24 \
MINLEN:50


fastqc \
-t 35 \
-o ${FASTQC_OUTDIR}/ \
${OUT_DIR}/paired_output/${FILE_PATTERN}*.fastq.gz
```

<br>

## Run FastQC on metagenomes after read quality trimming with Trimmomatic
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=compute # Queue selection
#SBATCH --job-name=imo # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/guaymas_andreas_metagenomes_trimmomatic_fastqc_nov_2022_%A-%a.log # Standard output/error
#SBATCH --array=0-12
#SBATCH --qos=unlim

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/Guaymas_Basin_Subsurface

module load anaconda/5.1
source activate trimmomatic

ADAPTER_FASTA=$(echo "/vortexfs1/home/dgellermcgrath/.conda/envs/trimmomatic/share/trimmomatic-0.39-2/adapters/NexteraPE-PE.fa")

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/Guaymas_Basin_Subsurface")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas")

FASTQC_OUTDIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/fastqc_of_trimmed_andreas_read_files")

FILE_PATTERN=("1545B_13H_4_S5" "1545B_19F_3_S6" "1545B_1H_2_S1" "1545B_2H_3_S2" "1545B_32F_3_S7" "1545B_4_H3_S3" "1545B_8_H3_S4" "1547B_1H_3_S8" "1547B_2H_3_S9" "1547B_3H_3_S10" "1547B_5H_3_S11" "1547B_7H_3_S12" "1547B_9H_3_S13")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


fastqc \
-t 35 \
-o ${FASTQC_OUTDIR}/ \
${OUT_DIR}/paired_output/${FILE_PATTERN}*.fastq.gz
```

<br>

## Run FastX and FastQC on metagenomes after initial FastQC; trim low quality ends
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=fastx_qc # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/guaymas_metagenomes_fastx_and_fastqc_%A-%a.log # Standard output/error
#SBATCH --array=0-25
#SBATCH --qos=scavenger

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas

module load anaconda/5.1
source activate trimmomatic


IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas/paired_output")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas/fastx_final_trimmed")

FASTQC_OUTDIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/fastqc_of_trimmed_andreas_read_files/final_fastqc_after_fastx_trimming_nov_14_2022")

FILE_PATTERN=("1545B_13H_4_S5_paired_1.fastq" "1545B_13H_4_S5_paired_2.fastq" "1545B_19F_3_S6_paired_1.fastq" "1545B_19F_3_S6_paired_2.fastq" "1545B_1H_2_S1_paired_1.fastq" "1545B_1H_2_S1_paired_2.fastq" "1545B_2H_3_S2_paired_1.fastq" "1545B_2H_3_S2_paired_2.fastq" "1545B_32F_3_S7_paired_1.fastq" "1545B_32F_3_S7_paired_2.fastq" "1545B_4_H3_S3_paired_1.fastq" "1545B_4_H3_S3_paired_2.fastq" "1545B_8_H3_S4_paired_1.fastq" "1545B_8_H3_S4_paired_2.fastq" "1547B_1H_3_S8_paired_1.fastq" "1547B_1H_3_S8_paired_2.fastq" "1547B_2H_3_S9_paired_1.fastq" "1547B_2H_3_S9_paired_2.fastq" "1547B_3H_3_S10_paired_1.fastq" "1547B_3H_3_S10_paired_2.fastq" "1547B_5H_3_S11_paired_1.fastq" "1547B_5H_3_S11_paired_2.fastq" "1547B_7H_3_S12_paired_1.fastq" "1547B_7H_3_S12_paired_2.fastq" "1547B_9H_3_S13_paired_1.fastq" "1547B_9H_3_S13_paired_2.fastq")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}

#gunzip trimmed fastq files prior to fastx trimming; it doesn't work on gzipped files
# cd $IN_DIR
# gunzip *.gz

fastx_trimmer \
-f 15 \
-i "$IN_DIR"/"$FILE_PATTERN" \
-o "$OUT_DIR"/"$FILE_PATTERN"

pigz "$OUT_DIR"/"$FILE_PATTERN"

fastqc \
-t 35 \
-o "$FASTQC_OUTDIR"/ \
"$OUT_DIR"/"$FILE_PATTERN".gz
```

<br>

## Co-assemble 26 Guaymas basin drill core metagenomes using MEGAHIT
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=bigmem # Queue selection
#SBATCH --job-name=co # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=79 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=3000gb # Job memory request
#SBATCH --time=10-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/26_guaymas_metagenomes_assembly_coassembly_megahit_default_nov_14_2022_%j.log # Standard output/error
#SBATCH --qos=unlim

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/all_fastx_trimmed_reads
source activate megahit

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly")

megahit -t 79 -1 1545B_13H_4_S5_paired_1.fastq.gz,1545B_19F_3_S6_paired_1.fastq.gz,1545B_1H_2_S1_paired_1.fastq.gz,1545B_2H_3_S2_paired_1.fastq.gz,1545B_32F_3_S7_paired_1.fastq.gz,1545B_4_H3_S3_paired_1.fastq.gz,1545B_8_H3_S4_paired_1.fastq.gz,1546B-1H2_S1_L004_paired_1.fastq.gz,1546B-3H2_S2_L004_paired_1.fastq.gz,1547B-1H2_S3_L004_paired_1.fastq.gz,1547B_1H_3_S8_paired_1.fastq.gz,1547B_2H_3_S9_paired_1.fastq.gz,1547B_3H_3_S10_paired_1.fastq.gz,1547B_5H_3_S11_paired_1.fastq.gz,1547B_7H_3_S12_paired_1.fastq.gz,1547B-8H2_S4_L004_paired_1.fastq.gz,1547B_9H_3_S13_paired_1.fastq.gz,1548B-2H3_S5_L004_paired_1.fastq.gz,1548B-4H7_S6_L004_paired_1.fastq.gz,1548B-8H5_S7_L004_paired_1.fastq.gz,1549B-3H2_S8_L004_paired_1.fastq.gz,U1545B_34F3_S9_L004_paired_1.fastq.gz,U1545B_4H2_S18_L004_paired_1.fastq.gz,U1547B_2H2_S12_L004_paired_1.fastq.gz,U1547B_5H2_S11_L004_paired_1.fastq.gz,U1547B_9H2_S14_L004_paired_1.fastq.gz -2 1545B_13H_4_S5_paired_2.fastq.gz,1545B_19F_3_S6_paired_2.fastq.gz,1545B_1H_2_S1_paired_2.fastq.gz,1545B_2H_3_S2_paired_2.fastq.gz,1545B_32F_3_S7_paired_2.fastq.gz,1545B_4_H3_S3_paired_2.fastq.gz,1545B_8_H3_S4_paired_2.fastq.gz,1546B-1H2_S1_L004_paired_2.fastq.gz,1546B-3H2_S2_L004_paired_2.fastq.gz,1547B-1H2_S3_L004_paired_2.fastq.gz,1547B_1H_3_S8_paired_2.fastq.gz,1547B_2H_3_S9_paired_2.fastq.gz,1547B_3H_3_S10_paired_2.fastq.gz,1547B_5H_3_S11_paired_2.fastq.gz,1547B_7H_3_S12_paired_2.fastq.gz,1547B-8H2_S4_L004_paired_2.fastq.gz,1547B_9H_3_S13_paired_2.fastq.gz,1548B-2H3_S5_L004_paired_2.fastq.gz,1548B-4H7_S6_L004_paired_2.fastq.gz,1548B-8H5_S7_L004_paired_2.fastq.gz,1549B-3H2_S8_L004_paired_2.fastq.gz,U1545B_34F3_S9_L004_paired_2.fastq.gz,U1545B_4H2_S18_L004_paired_2.fastq.gz,U1547B_2H2_S12_L004_paired_2.fastq.gz,U1547B_5H2_S11_L004_paired_2.fastq.gz,U1547B_9H2_S14_L004_paired_2.fastq.gz -o ${OUT_DIR}
```

<br>

## Create bins from the metagenomic co-assembly using MaxBin2
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=compute # Queue selection
#SBATCH --job-name=mxbc # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=35 #
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=7-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/maxbin2_andreas_coassembly_binning_nov_26_2022_%j.log# Standard output/error
#SBATCH --qos=unlim

source activate maxbin2_final

# create bins with maxbin2 -- on the original megabit coassembly

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly")

BAM_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/bam_files")

BIN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/maxbin2")

run_MaxBin.pl \
-contig  "$IN_DIR"/final.contigs.fa \
-abund "$BAM_DIR"/andreas_coassembly_contig_coverage_from_default_coverM_settings_nov_25_2022.txt \
-out "$BIN_DIR" \
-thread 35 \
-markerset 40
```

<br>

## Create bins from the metagenomic co-assembly using MetaBat2
```{bash, eval=FALSE, include=TRUE}
conda activate metabat2

#command to create coverage file from bam files for metabat2
jgi_summarize_bam_contig_depths --outputDepth summaryFile.txt *.bam 

# mv summaryFile.txt andreas_coassembly_contig_coverage_from_default_coverM_settings_nov_25_2022.txt


# create bins with metabat2 -- on the original metabat2 coassembly - made with default megahit settings

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly")

BAM_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/bam_files")

BIN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/metabat2")

metabat2 \
--saveCls \
--numThreads 36 \
-i  "$IN_DIR"/final.contigs.fa \
-a "$BAM_DIR"/andreas_coassembly_contig_coverage_from_default_coverM_settings_nov_25_2022.txt \
-o "$BIN_DIR"/MAG_ \
--unbinned

```

<br>

## Create bins from metagenomic co-assembly using CONCOCT
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=samtools # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 #
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/samtools_sort_and_index_array_%A-%a.log# Standard output/error
#SBATCH --array=0-25
#SBATCH --qos=scavenger

# command to print out patterns used in FILE_PATTERN array below
# ls *.bam | sed -r "s/.*.fa.(.*).fastq.*/\"\1\"/" | tr '\n' ' '

source activate concoct

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/nov28_bam_files

FILE_PATTERN=("1545B_13H_4_S5_paired_1" "1545B_19F_3_S6_paired_1" "1545B_1H_2_S1_paired_1" "1545B_2H_3_S2_paired_1" "1545B_32F_3_S7_paired_1" "1545B_4_H3_S3_paired_1" "1545B_8_H3_S4_paired_1" "1546B-1H2_S1_L004_paired_1" "1546B-3H2_S2_L004_paired_1" "1547B-1H2_S3_L004_paired_1" "1547B_1H_3_S8_paired_1" "1547B_2H_3_S9_paired_1" "1547B_3H_3_S10_paired_1" "1547B_5H_3_S11_paired_1" "1547B_7H_3_S12_paired_1" "1547B-8H2_S4_L004_paired_1" "1547B_9H_3_S13_paired_1" "1548B-2H3_S5_L004_paired_1" "1548B-4H7_S6_L004_paired_1" "1548B-8H5_S7_L004_paired_1" "1549B-3H2_S8_L004_paired_1" "U1545B_34F3_S9_L004_paired_1" "U1545B_4H2_S18_L004_paired_1" "U1547B_2H2_S12_L004_paired_1" "U1547B_5H2_S11_L004_paired_1" "U1547B_9H2_S14_L004_paired_1")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


# create sorted bam file
samtools sort final.contigs.fa."$FILE_PATTERN".fastq.gz.bam -o final.contigs.fa."$FILE_PATTERN".fastq.gz.sorted.bam


# create indexed bam file
samtools index final.contigs.fa."$FILE_PATTERN".fastq.gz.sorted.bam final.contigs.fa."$FILE_PATTERN".fastq.gz.sorted.bam.bai

-----
-----
# done interactively on a compute node:

conda activate concoct

BAM_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/nov28_bam_files")

CONCOCT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/concoct")

# cut up contigs into smaller parts
cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly
cut_up_fasta.py final.contigs.fa -c 10000 -o 0 --merge_last -b contigs_10K.bed > contigs_10K.fa

# Generate table with coverage depth information per sample and subcontig. This step assumes the directory 'mapping' contains sorted and indexed bam files where each sample has been mapped against the original contigs.

conda deactivate
conda activate concoct

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly
concoct_coverage_table.py contigs_10K.bed "$BAM_DIR"/*.sorted.bam > "$CONCOCT_DIR"/metadata/nov28_concoct_coverage_table.tsv

# run concoct
NUMEXPR_MAX_THREADS=35
concoct --composition_file contigs_10K.fa --coverage_file "$CONCOCT_DIR"/metadata/nov28_concoct_coverage_table.tsv -b "$CONCOCT_DIR"/ -t 35

# Merge subcontig clustering into original contig clustering
merge_cutup_clustering.py "$CONCOCT_DIR"/clustering_gt1000.csv > "$CONCOCT_DIR"/clustering_merged.csv

# Extract bins as individual FASTA
mkdir "$CONCOCT_DIR"/fasta_bins
extract_fasta_bins.py final.contigs.fa "$CONCOCT_DIR"/clustering_merged.csv --output_path "$CONCOCT_DIR"/fasta_bins
```

<br>

## Aggregate and dereplicate bins using Das Tool
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=dasTool # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=35 #
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=4-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/das_tool_nov30_2022_%j.log# Standard output/error
#SBATCH --qos=scavenger

# following commands were run interactively on a compute node:

#conda activate das_tool

#Fasta_to_Contig2Bin.sh -e fa -i /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/concoct/fasta_bins > concoct_contig2bin.tsv

# truncated contig headers & contig2bin contig headers; was getting error with DAS Tool

# cp final.contigs.fa truncatedHeaders_copy_final.contigs.faq

#cat truncatedHeaders_copy_final.contigs.fa | sed -r 's/(^>.*) flag.*/\1/' > tmp.fa && mv -f tmp.fa truncatedHeaders_copy_final.contigs.fa

#cat concoct_contig2bin.tsv | sed -r 's/(^k.*) flag.*len=[0-9]+\s+([0-9]+)/\1\t\2/' > tmp.tsv && mv -f tmp.tsv concoct_contig2bin.tsv

#not changing metabat2 and maxbin2 contig2bin files. by default it has kept the k_[0-9]+ part of the megahit coassembly contig headers, and gotten rid of everything after the first space. that's probably why DAS Tool was failing earlier...

#Fasta_to_Contig2Bin.sh -i /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/metabat2 -e fna > metabat2_contig2bin.tsv

#Fasta_to_Contig2Bin.sh -i /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/maxbin2 -e fasta > maxbin2_contig2bin.tsv

source activate das_tool

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/dasTool_analysis/contig2bin_inputs

# create refined bins with DAS_tool

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/dasTool_analysis/final_refined_bins_november_2022")


DAS_Tool \
-i concoct_contig2bin.tsv,metabat2_contig2bin.tsv,maxbin2_contig2bin.tsv \
-l concoct,metabat,maxbin \
-c "$IN_DIR"/truncatedHeaders_copy_final.contigs.fa \
-o "$OUT_DIR"/Guaymas_MAG_P_ \
--write_bins \
-t 35
```

<br>

## Classify taxonomy of Das Tool bins
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=gtdb-tk # taxonomic assessent of DAS Tool Guaymas bins
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=35 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=187gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/andreas_coassembly_gtdbtk_on_guaymasDasToolMegahitBins_%j.log # Standard output/error
#SBATCH --qos=scavenger

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385

source activate gtdbtk

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/gtdbtk")

gtdbtk classify_wf \
--cpus 35 \
--pplacer_cpus 1 \
--genome_dir "$IN_DIR" \
-x fa \
--force \
--out_dir "$OUT_DIR"
```

<br>

## Run CoverM on MAGs to calculate their relative abundances in all metagenomic samples
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=coverM # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=35 #
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/dec_1_2022_andreas_dasTool_mags_relAbun_coverM_array_%A-%a.log# Standard output/error
#SBATCH --array=0-25
#SBATCH --qos=scavenger

source activate coverm

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/all_fastx_trimmed_reads

FILE_PATTERN=("1545B_13H_4_S5" "1545B_19F_3_S6" "1545B_1H_2_S1" "1545B_2H_3_S2" "1545B_32F_3_S7" "1545B_4_H3_S3" "1545B_8_H3_S4" "1546B-1H2_S1_L004" "1546B-3H2_S2_L004" "1547B-1H2_S3_L004" "1547B_1H_3_S8" "1547B_2H_3_S9" "1547B_3H_3_S10" "1547B_5H_3_S11" "1547B_7H_3_S12" "1547B-8H2_S4_L004" "1547B_9H_3_S13" "1548B-2H3_S5_L004" "1548B-4H7_S6_L004" "1548B-8H5_S7_L004" "1549B-3H2_S8_L004" "U1545B_34F3_S9_L004" "U1545B_4H2_S18_L004" "U1547B_2H2_S12_L004" "U1547B_5H2_S11_L004" "U1547B_9H2_S14_L004")

GENOME_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/rel_abun_files")


FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}

# always double check whether fastq files are gzipped or not

coverm genome \
--coupled "$FILE_PATTERN"_paired_1.fastq.gz "$FILE_PATTERN"_paired_2.fastq.gz \
--genome-fasta-directory "$GENOME_DIR" \
--mapper bwa-mem \
--methods relative_abundance \
--min-read-percent-identity 95 \
--min-read-aligned-percent 50 \
--min-covered-fraction 0 \
--exclude-supplementary \
--genome-fasta-extension fa \
--threads 35 \
--output-file "$OUT_DIR"/"$FILE_PATTERN"_refined_bin_coverages.tsv
```

<br>

## Call translated genes with Prodigal; annotate .faa output files with KofamScan command line tool
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=pr_kf # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=187gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/dec_5_2022_prodigal_kofamscan_on_guaymas_mags_array_%A-%a.log# Standard output/error
#SBATCH --array=0-141
#SBATCH --qos=scavenger


# load conda environment containing Prodigal and Kofamscan
source activate upd-kofam

# kofamscan hmm profiles directory
HMM_DIR=$(echo "/vortexfs1/omics/pachiadaki/dgellermcgrath/kofam_hmm_profiles")

# input directories
MAG_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")

# output directories
PROD_FNA_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/prodigal/fna")
PROD_FAA_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/prodigal/faa")
KOFAM_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/kofamscan")


#mag name file pattern
FILE_PATTERN=("MAG_130" "MAG_51" "MAG_244" "MAG_263" "MAG_12" "MAG_229" "145" "MAG_292" "MAG_117" "174" "112_sub" "MAG_305" "MAG_284" "MAG_334" "26_sub" "MAG_119" "280" "MAG_33" "MAG_116" "MAG_53" "MAG_249" "MAG_210" "MAG_315" "289" "MAG_155" "MAG_46" "MAG_153_sub" "MAG_144" "MAG_234" "MAG_200" "73_sub" "MAG_162" "MAG_193" "MAG_126" "275_sub" "MAG_328" "MAG_141" "259_sub" "MAG_313" "MAG_149_sub" "MAG_214" "MAG_289" "MAG_133" "MAG_247" "MAG_273_sub" "MAG_206" "16_sub" "MAG_128" "MAG_280_sub" "MAG_56_sub" "MAG_39" "MAG_226" "MAG_140" "MAG_172" "270_sub" "MAG_142" "MAG_327_sub" "MAG_326" "2" "MAG_42_sub" "MAG_94" "58" "131_sub" "245_sub" "238_sub" "MAG_13" "MAG_332_sub" "97_sub" "MAG_171" "163" "MAG_277" "MAG_235" "MAG_80_sub" "MAG_58" "MAG_18" "MAG_199" "MAG_114" "MAG_222_sub" "MAG_269_sub" "123_sub" "MAG_121" "169_sub" "MAG_217_sub" "MAG_74" "MAG_316" "MAG_224" "103_sub" "MAG_314" "223_sub")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


prodigal \
-i "$MAG_DIR"/"$FILE_PATTERN".fa \
-a "$PROD_FAA_DIR"/"$FILE_PATTERN".faa \
-d "$PROD_FNA_DIR"/"$FILE_PATTERN".fna


exec_annotation \
-o "$KOFAM_DIR"/"$FILE_PATTERN"-ko.tsv \
--cpu 36 \
--tmp-dir "$KOFAM_DIR"/"$FILE_PATTERN" \
-f detail-tsv \
-p "$HMM_DIR"/profiles/prokaryote.hal \
-k "$HMM_DIR"/ko_list "$PROD_FAA_DIR"/"$FILE_PATTERN".faa
```

<br>

## Annotate the MAGs using Prokka
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=prokka # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=187gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/feb_3_2023_prokka_annotation_of_guaymas_mags_array_%A-%a.log# Standard output/error
#SBATCH --array=0-88
#SBATCH --qos=scavenger


# directory containing the Guaymas MAGs
MAG_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")


# directory for all dram annotation output
OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/prokka/annotations")


# load prokka conda environment
source activate prokka


# bash array of mag name identifiers
FILE_PATTERN=("MAG_130" "MAG_51" "MAG_244" "MAG_263" "MAG_12" "MAG_229" "145" "MAG_292" "MAG_117" "174" "112_sub" "MAG_305" "MAG_284" "MAG_334" "26_sub" "MAG_119" "280" "MAG_33" "MAG_116" "MAG_53" "MAG_249" "MAG_210" "MAG_315" "289" "MAG_155" "MAG_46" "MAG_153_sub" "MAG_144" "MAG_234" "MAG_200" "73_sub" "MAG_162" "MAG_193" "MAG_126" "275_sub" "MAG_328" "MAG_141" "259_sub" "MAG_313" "MAG_149_sub" "MAG_214" "MAG_289" "MAG_133" "MAG_247" "MAG_273_sub" "MAG_206" "16_sub" "MAG_128" "MAG_280_sub" "MAG_56_sub" "MAG_39" "MAG_226" "MAG_140" "MAG_172" "270_sub" "MAG_142" "MAG_327_sub" "MAG_326" "2" "MAG_42_sub" "MAG_94" "58" "131_sub" "245_sub" "238_sub" "MAG_13" "MAG_332_sub" "97_sub" "MAG_171" "163" "MAG_277" "MAG_235" "MAG_80_sub" "MAG_58" "MAG_18" "MAG_199" "MAG_114" "MAG_222_sub" "MAG_269_sub" "123_sub" "MAG_121" "169_sub" "MAG_217_sub" "MAG_74" "MAG_316" "MAG_224" "103_sub" "MAG_314" "223_sub")


# pick one mag name for the current array iteration
FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


# run prokka
prokka \
--outdir "$OUT_DIR"/"$FILE_PATTERN" \
--prefix "$FILE_PATTERN" \
--cpus 35 \
"$MAG_DIR"/"$FILE_PATTERN".fa
```

<br>

## Map metatranscriptomic reads to the MAGs
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=compute # Queue selection
#SBATCH --job-name=coverM # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/feb_4_2023_coverm_mag_counts_of_metaT_reads_mapped_array_%A-%a.log# Standard output/error
#SBATCH --array=0-18
#SBATCH --qos=unlim

source activate coverm

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/transcriptomes/Fastx_output")

MAG_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/metaT/reads_mapped_per_genome")

FILE_PATTERN=("1545_11H3" "1545_1H2" "1545_4H3" "1546_12H2" "1546_1H2" "1546_3H2" "1547_1H2" "1547_5H2" "1547_9H2" "1548_1H2" "1548_4H7" "1549_1H2" "1549_3H2" "1550_1H2" "1550_3H2" "1551_1H2" "1551_3H2" "1552_1H2" "1552_3H4")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


coverm \
genome \
--coupled "$IN_DIR"/"$FILE_PATTERN"_R1_paired_fastx.fastq "$IN_DIR"/"$FILE_PATTERN"_R2_paired_fastx.fastq \
--genome-fasta-directory "$MAG_DIR" \
--genome-fasta-extension fa \
--mapper bwa-mem \
--methods count relative_abundance mean trimmed_mean rpkm tpm \
--min-covered-fraction 0 \
--min-read-percent-identity 95 \
--min-read-aligned-percent 50 \
--exclude-supplementary \
--threads 36 \
-o "$OUT_DIR"/"$FILE_PATTERN"_counts_reads_mapped_per_genome.tsv
```
