---
title: "Deep Ocean Drilling Reveals Bacterial and Archaeal Metagenome-assembled Genomes Shaped by Thermal and Geochemical Gradients in Deep Sediments at Contrasting Sites in Guaymas Basin"
author: David Geller-McGrath
date: February 16, 2023
output:
  html_document:
    df_print: paged
    toc: true # create a table of contents
    toc_depth: 5 # up to five depths of headings in the toc
    theme: united # select the document theme
    syntax: tango # select syntax highlight theme
---

<br>

This code was used for the analyses of Guaymas Basin metagenomes and metatranscriptomes for the 2023 publication "Deep Ocean Drilling Reveals Bacterial and Archaeal Metagenome-assembled Genomes Shaped by Thermal and Geochemical Gradients in Deep Sediments at Contrasting Sites in Guaymas Basin" by Geller-McGrath et al. Some sections of these analyses were done on a compute node of the Poseidon High Performance Computing (HPC) cluster based at the Woods Hole Oceanographic Institute (WHOI). All of the analyses done here using R can be run locally; HPC processing was used to speed up computations.

<br>

Load required R libraries

```{r include=TRUE, message=FALSE, warning=FALSE}
library(tidyverse)
library(vroom)
library(ComplexHeatmap)
library(gridtext)
library(RColorBrewer)
library(ecodist)
library(vegan)
library(ape)
library(ggrepel)
```

<br>

## Figure 2. MAG relative abundances
```{r echo=TRUE, message=FALSE, warning=FALSE}
# Figure 2
final_filtered_andreas_mag_summary_dec_2022 = 
  readRDS('rds_files/final_filtered_andreas_mag_summary_dec_2022.rds')

guay_relAbun_files = system('ls flat_files/mag_relative_abundance_coverM/new_rel_abun_dec_2022/*.tsv',
                            intern = TRUE)

guay_relAbun_tbl = guay_relAbun_files |>
  map(~ read_tsv(.x, show_col_types = FALSE))

guay_relAbun_tbl = guay_relAbun_tbl |>
  reduce(left_join, by = 'Genome')


guay_relAbun_tbl = guay_relAbun_tbl |>
  rename_with(~ str_replace(.x, '(.*)_(.*)_S\\d+.*', '\\1\\2')) |>
  rename_with(~ str_replace(.x, '(.*)-(.*)_S\\d+.*', '\\1_\\2')) |>
  rename_with(~ str_replace(.x, '^U(.*B)(.*)', '\\1_\\2')) |>
  rename(bin = Genome) |>
  slice(-1) # remove the unmapped % information; won't be used in the heatmap

guay_relAbun_tbl = guay_relAbun_tbl |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, bin),
            by = 'bin') |>
  relocate(mag_name, .before = 1) |>
  filter(!is.na(mag_name)) |>
  select(-bin)

guayMagRelAbun_rowAnno = guay_relAbun_tbl |>
  select(mag_name) |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, phylum, class),
            by = 'mag_name') |>
  mutate(phylum = case_when(
    class == 'Alphaproteobacteria' ~ 'Alphaproteobacteria',
    is.na(phylum) ~ 'Unclassified',
    TRUE ~ phylum)) |>
  rename(`Phylum/Class` = phylum) |>
  select(-class) |>
  column_to_rownames('mag_name')


guayMagRelPalette = list(
  `Phylum/Class` = c(
  'Candidatus Niteriota'	=	'wheat1',
  'Acidobacteriota'	=	'tomato',
  'Aenigmatarchaeota'	=	'orchid1',
  'Aerophobota'	=	'thistle1',
  'Alphaproteobacteria'	=	'slateblue1',
  'Asgardarchaeota'	=	'aquamarine',
  'Atribacterota'	=	'gold',
  'Bipolaricaulota'	=	'darkred',
  'Chloroflexota'	=	'darkviolet',
  'Cloacimonadota'	=	'khaki',
  'Desulfobacterota'	=	'firebrick1',
  'Hadarchaeota'	=	'gainsboro',
  'Marinisomatota'	=	'springgreen',
  'Methanobacteriota_B'	=	'chocolate4',
  'Planctomycetota'	=	'yellowgreen',
  'Candidatus Rosutiota'	=	'deepskyblue',
  'Thermoplasmatota'	=	'#666666',
  'Thermoproteota'	=	'#B89B74',
  'Unclassified'	=	'lightsteelblue1',
  'WOR-3'	=	'ivory',
  "Zixibacteria"	=	'#2A7FB7'),
  
  `Depth (mbsf)` = c('white', 'black'),

  Site = brewer.pal(8, 'Pastel2')[3:7] |>
    set_names('U1545B', 'U1546B', 'U1547B', 'U1548B', 'U1549B'))


guay_relAbun_plot_data =
  guay_relAbun_tbl |>
  mutate(across(2:last_col(), ~ log1p(.x))) |>
  left_join(
    guayMagRelAbun_rowAnno |>
      rownames_to_column('mag_name'),
    by = 'mag_name') |>
  relocate(`Phylum/Class`, .after = 1) |> 
  group_by(`Phylum/Class`) |>
  mutate(group_total = n()) |>
  ungroup() |>
  arrange(desc(group_total), `Phylum/Class`) |> 
  select(-c(`Phylum/Class`, group_total)) |>
  column_to_rownames('mag_name')


final_guay_relAbun_colAnno = read_tsv('flat_files/metagenome_site_metadata.txt')
final_guay_relAbun_colAnno$Sample[12] = 'U1545B-1_25.8'
final_guay_relAbun_colAnno$Sample[13] = 'U1545B-2_25.8'
final_guay_relAbun_colAnno = final_guay_relAbun_colAnno |> 
  column_to_rownames('Sample')

metagenome_sample_metadata = read_tsv('flat_files/metagenome_sample_name_metadata.tsv')

    
guay_relAbun_plot_data = guay_relAbun_plot_data |> 
  rownames_to_column('mag_name') |> 
  pivot_longer(
    cols = 2:last_col(),
    names_to = 'Sample',
    values_to = 'value') |> 
  left_join(metagenome_sample_metadata, by = 'Sample') |>
  mutate(Sample = Sample |> str_replace_all(
    c('1545B_4H2' = '1545B-1_4H2', '1545B_4H3' = '1545B-2_4H3')),
    Sample = Sample |> str_replace('(.*_).*', paste0('U\\1', `Depth (mbsf)`))) |> 
  select(-`Depth (mbsf)`) |> 
  pivot_wider(
    names_from = Sample,
    values_from = value) |> 
  unnest(everything()) |> 
  relocate(c(mag_name, rownames(final_guay_relAbun_colAnno)))


guay_row_anno = guay_relAbun_plot_data |> 
  select(mag_name) |> 
  left_join(final_filtered_andreas_mag_summary_dec_2022 |> 
              select(c(mag_name, phylum, class)),
            by = 'mag_name') |> 
  mutate(phylum = case_when(
    phylum == 'TA06_A' ~ 'Candidatus Rosutiota',
    phylum == '4484-113' ~ 'Candidatus Niteriota',
    TRUE ~ phylum)) |> 
  mutate(`Phylum/Class` = if_else(
    class == 'Alphaproteobacteria', class, phylum)) |> 
  select(-c(phylum, class)) |>
  column_to_rownames('mag_name') |> 
  mutate(`Phylum/Class` = as_factor(`Phylum/Class`))
  

guay_col_anno = tibble(sample = guay_relAbun_plot_data |> 
  select(-mag_name) |> 
  colnames()) |> 
  mutate(`Depth (mbsf)` = sample,
         `Depth (mbsf)` = `Depth (mbsf)` |> 
           str_replace('.*_(.*)', '\\1') |> 
           as.numeric()) |> 
  column_to_rownames('sample')

guay_relAbun_complexHeatmap =
  guay_relAbun_plot_data |>
  column_to_rownames('mag_name') |> 
  as.matrix() |>
  ComplexHeatmap::pheatmap(
    name = 'Relative abundance
(% total reads mapped)',
main = 'Guaymas MAG relative abundance',
show_colnames = TRUE,
annotation_row = guay_row_anno,
annotation_col = guay_col_anno,
annotation_colors = guayMagRelPalette,
cluster_rows = FALSE,
cluster_cols = FALSE,
show_rownames = FALSE,
border_color = 'grey30',
treeheight_row = 0,
treeheight_col = 10,
color = c('black', viridis::viridis(n = 50)),
legend_breaks = 0:3,
legend_labels = gt_render(c('0', sprintf('%.1f', round(expm1(1:3), 1)))),
show_row_dend = FALSE,
show_column_dend = TRUE)

draw(guay_relAbun_complexHeatmap,
     merge_legend = TRUE)

```

<br> 

## Figure 3. MAG transcript abundances
```{r echo=TRUE, message=FALSE, warning=FALSE}
# Figure 3

metaT_metadata = readRDS('rds_files/metaT_metadata')

genome_metaT_files = system('ls ~/Documents/guaymas_analysis/flat_files/reads_mapped_per_genome/*', intern = TRUE)

read_genome_metaT_data = function(.coverM_genome_TPM_file) {
  read_tsv(.coverM_genome_TPM_file, show_col_types = FALSE) |>
    rename(bin = Genome) |>
    filter(if_any(.cols = 2, ~ !is.na(.x))) |> #filter by col number, not col name
    select(1, 7) |> # select bin name col & genome read mapping counts col (normalized in TPM format)
    rename_with(~ str_replace(.x, '_R1_paired_fastx.fastq TPM', ''))
}

mag_data_cols = c('mag_name', 'phylum', 'class', 'order', 'completeness')

genome_metaT_data = genome_metaT_files |>
  map(function(.file) {read_genome_metaT_data(.file)}) |>
  reduce(left_join, by = 'bin') |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |>
              select(mag_name, bin, phylum, class, order, completeness),
            by = 'bin') |>
  select(-bin) |>
  relocate(all_of(mag_data_cols)) |>
  filter(!is.na(mag_name)) |>
  arrange(phylum) |>
  mutate(phylum = case_when(
    phylum == 'TA06_A' ~ 'Candidatus Rosutiota',
    phylum == '4484-113' ~ 'Candidatus Niteriota',
    TRUE ~ phylum)) |>
  mutate(`Phylum/Class` = if_else(class == 'Alphaproteobacteria', class, phylum)) |>
  relocate(`Phylum/Class`, .after = 1) |>
  group_by(`Phylum/Class`) |>
  add_tally(n(), name = 'total') |>
  ungroup() |>
  relocate(total) |>
  arrange(desc(total), phylum, order) |>
  select(-total)
#select(-c(class, order, completeness))

genome_metaT_rowAnno = genome_metaT_data |>
  select(mag_name, `Phylum/Class`) |>
  mutate(`Phylum/Class` = as_factor(`Phylum/Class`)) |>
  column_to_rownames('mag_name')

genome_metaT_colAnno = metaT_metadata |>
  column_to_rownames('sample') |>
  rename(
    `Depth (mbsf)` = depth,
    Site = site) |>
  select(-Site)

genome_metaT_heat_data = genome_metaT_data |>
  select(-c(`Phylum/Class`, phylum, class, order, completeness)) |>
  column_to_rownames('mag_name')




metaT_mag_row_order = guay_relAbun_plot_data |>
  pull(mag_name)

genome_metaT_colAnno_onlyMetagenomeSites = genome_metaT_colAnno |>
  rownames_to_column('sample') |>
  filter(str_detect(sample, '1545|1546|1547|1548|1549')) |>
  column_to_rownames('sample')

genome_metaT_heat_onlyMetagenomeSites = genome_metaT_heat_data |>
  relocate(metaT_metadata$sample) |>
  select(matches('1545.*|1546.*|1547.*|1548.*|1549.*'))


genome_metaT_rowAnno2 = genome_metaT_rowAnno |>
  rownames_to_column('mag_name') |>
  arrange(factor(mag_name, levels = metaT_mag_row_order)) |>
  column_to_rownames('mag_name') |> 
  mutate(`Phylum/Class` = fct_inorder(`Phylum/Class`))
  


metaT_heatmap = genome_metaT_heat_onlyMetagenomeSites |>
  rownames_to_column('mag_name') |>
  as_tibble() |>
  pivot_longer(
    cols = 2:last_col(),
    names_to = 'sample',
    values_to = 'value') |>
  left_join(metaT_metadata, by = 'sample') |>
  mutate(
    depth = round(depth, digits = 1),
    site = paste0('U', site, '_', depth)) |>
  select(-c(sample, depth)) |>
  pivot_wider(
    names_from = site,
    values_from = value) |>
  arrange(factor(mag_name, levels = metaT_mag_row_order)) |>
  column_to_rownames('mag_name') |>
  as.matrix() |>
  ComplexHeatmap::pheatmap(
    name = 'Transcripts per million',
    main = 'MAG metatranscriptomic read recruitment',
    show_colnames = TRUE,
    show_rownames = FALSE,
    annotation_col = genome_metaT_colAnno_onlyMetagenomeSites,
    annotation_row = genome_metaT_rowAnno2,
    annotation_colors = guayMagRelPalette,
    cluster_rows = FALSE,
    cluster_cols = FALSE,
    border_color = 'grey30',
    treeheight_row = 0,
    treeheight_col = 10,
    color = c('black', viridis::viridis(n = 50)),
    # legend_breaks = 0:3,
    # legend_labels = gt_render(
    #   c('0', sprintf('%.2f', round(expm1(1:3), 2)))
    # ),
    show_row_dend = FALSE,
    show_column_dend = TRUE
  )

draw(metaT_heatmap, merge_legend = TRUE)
```

<br> 

## Figure 4. Non-metric multidimensional scaling (nMDS) ordination plot of metagenomes reconstructed from sites U1545B, U1546B, U1547B, U1548B, U1549B and environmental parameters with a significant effect on microbial community composition
```{r echo=TRUE, message=FALSE, warning=FALSE}
# Figure 4

env_metadata2 = readRDS('rds_files/env_metadata2.rds')

guay_tpm_files = system('ls flat_files/mag_relative_abundance_coverM/new_tpm_abun_dec_2022/*.tsv', intern = TRUE)

guay_tpm_tbl = guay_tpm_files |>
  map(~ read_tsv(.x, show_col_types = FALSE)) |>
  reduce(left_join, by = 'Genome')


guay_tpm_tbl = guay_tpm_tbl |>
  # rename_with(
  #   ~ str_replace(.x, '(.*)-(.*)_S\\d_L004_paired_1.fastq Relative Abundance \\(%\\)', '\\1\\2'), # the literal parentheses from "(%)" need to be marked with \\
  #   .cols = 2:9
  # ) |>
  # rename_with(
  #   ~ str_replace(.x, '(.*)_(.*)_S.*', '\\1\\2'),
  #   .cols = 10:last_col()
  # ) |>

  rename_with(~ str_replace(.x, '(.*)_(.*)_S\\d+.*', '\\1\\2')) |> #colnames()
  rename_with(~ str_replace(.x, '(.*)-(.*)_S\\d+.*', '\\1_\\2')) |>  #colnames()
  rename_with(~ str_replace(.x, '^U(.*B)(.*)', '\\1_\\2')) |>
  rename(bin = Genome)


guay_tpm_tbl = guay_tpm_tbl |>
  left_join(final_filtered_andreas_mag_summary_dec_2022 |> select(mag_name, bin), by = 'bin') |>
  relocate(mag_name, .before = 1) |>
  filter(!is.na(mag_name)) |>
  select(-bin)


guay_tpm_pcoa_data = guay_tpm_tbl |>
      pivot_longer(
        cols = 2:last_col(),
        names_to = 'sample',
        values_to = 'value') |> 
      left_join(metagenome_sample_metadata |>
                  rename(sample = Sample),
                by = 'sample') |> 
      mutate(sample = sample |> str_replace_all(
        c('1545B_4H2' = '1545B-1_4H2', '1545B_4H3' = '1545B-2_4H3')),
        sample = sample |> str_replace('(.*_).*', paste0('U\\1', `Depth (mbsf)`))) |> 
      select(-`Depth (mbsf)`) |> 
      pivot_wider(
        names_from = sample,
        values_from = value) |> 
      unnest(everything()) |> 
      relocate(c(mag_name, rownames(final_guay_relAbun_colAnno))) |>
      pivot_longer(
        cols = 2:last_col(),
        names_to = 'sample',
        values_to = 'relative_abundance') |>
      pivot_wider(
        names_from = mag_name,
        values_from = relative_abundance) |>
      column_to_rownames('sample')


guay_tpm_pcoa_data_FINAL = guay_tpm_pcoa_data |>
  rownames_to_column('sample') |>
  pivot_longer(
    2:last_col(),
    names_to = 'mag',
    values_to = 'tpm'
  ) |>
  pivot_wider(
    names_from = sample,
    values_from = tpm
  ) |>
  column_to_rownames('mag') |>
  decostand(method = 'hellinger') |>
  rownames_to_column('mag') |>
  as_tibble() |>
  pivot_longer(
    2:last_col(),
    names_to = 'sample',
    values_to = 'tpm'
  ) |>
  pivot_wider(
    names_from = mag,
    values_from = tpm)



set.seed(123)
dist = guay_tpm_pcoa_data_FINAL |>
  #mutate(sample = sample |> str_replace('(.*)_.*', '\\1')) |>
  column_to_rownames('sample') |>
  vegdist(method = 'bray')


nmds_result_FINAL = metaMDS(
  dist,
  trymax = 10000,
  autotransform = TRUE)


nmds_scores_FINAL = scores(nmds_result_FINAL, "site") |>
  as.data.frame() %>%
  mutate(sample = rownames(.)) |>
  as_tibble() |>
  left_join(env_metadata2 |>
              #select(`Depth (mbsf)`) |>
              rownames_to_column('sample'),
              #rename(depth = `Depth (mbsf)`),
            by = 'sample') |>
  mutate(site = str_replace(
    sample, '(\\d{4}[:upper:]).*', '\\1'))


FINAL_nmds_envFit = envfit(
  nmds_result_FINAL,
  env_metadata2,
  na.rm = TRUE,
  permutations = 10000)


FINAL_nmds_envScores = as.data.frame(scores(FINAL_nmds_envFit, display = 'vectors')) %>%
  mutate(env_variable = rownames(.),
         pvalue = FINAL_nmds_envFit$vectors$pvals) |>
  filter(pvalue <= 0.05)

FINAL_nmds_envScores


rshift = function(r, theta, a=0.03, b=0.07) {
  r + a + b*abs(cos(theta))
}


# Calculate shift
FINAL_env.arrows = FINAL_nmds_envScores %>%
  mutate(r = sqrt(NMDS1^2 + NMDS2^2),
         theta = atan2(NMDS2,NMDS1),
         rnew = rshift(r, theta),
         xnew = rnew*cos(theta),
         ynew = rnew*sin(theta))


nmds_scores_FINAL = nmds_scores_FINAL |>
  rename(depth = `Depth (mbsf)`) |> 
  mutate(depth_bin = case_when(
    depth < 15 ~ 'Shallow (0-15 mbsf)',
    depth >= 15 & depth < 60 ~ 'Intermediate (15-60 mbsf)',
    depth >= 60 ~ 'Deep (> 60 mbsf)'),
    Site = paste0('U', Site))


FINAL_nmds_plot_env = nmds_scores_FINAL |>

  ggplot(aes(x = NMDS1, y = NMDS2)) +
  geom_point(aes(color = site,
                 shape = factor(
                   depth_bin,
                   levels = c('Shallow (0-15 mbsf)',
                              'Intermediate (15-60 mbsf)',
                              'Deep (> 60 mbsf)'))),
             size = 3,
             alpha = 0.5) +
  scale_color_manual(values = c(
    'U1545B' = 'purple',
    'U1546B' = 'blue',
    'U1547B' = 'red',
    'U1548B' = 'goldenrod1',
    'U1549B' = 'green'
  )) +
  geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2),
               data = FINAL_nmds_envScores,
               size = 1,
               alpha = 0.25,
               color = "grey30",
               arrow = arrow(length = unit(0.25, 'cm'))) +
  geom_text(
    data = FINAL_env.arrows,
    aes(x = xnew, y = ynew,
        label = env_variable),
    cex = 4.7) +#,
  theme(
    axis.title = element_text(size = 14,
                              face = "bold",
                              colour = "grey30"),
    axis.text = element_text(size = 14),
    panel.background = element_blank(),
    panel.border = element_rect(fill = NA,
                                colour = "grey30"),
    legend.margin = margin(b = -5),
    legend.position = 'top',
    legend.box = 'vertical',
    legend.key = element_blank(),
    text = element_text(size = 14),
    legend.title = element_text(size = 11,
                                face = "bold", colour = "grey30"),
    legend.text = element_text(size = 10,
                               colour = "grey30")) +
  labs(
    shape = 'Depth',
    color = 'Sampling site',
    x = '\nNMDS1',
    y = 'NMDS2\n'
  ) +
  geom_label(x = -0.6, y = 1.2,
             label = 'Stress = 0.057',
             label.size = NA)

FINAL_nmds_plot_env
```
<br>

## Figure 5b. Taxonomic order-specific relative abundance of Chloroflexota and Thermoproteota MAGs in metagenomic samples from drill sites U1545B and U1547B, and corresponding geochemical parameter measurements
```{r}
# Figure 5

env_params_1546_1547_comparison_data =
  env_metadata2 |>
  mutate(Site = paste0('U', Site)) |> 
  rename(site = Site) |> 
  rownames_to_column('sample') |>
  as_tibble() |>
  select(-c(5,6,7,9,12,13,19:22)) |>
  filter(site %in% c('U1545B', 'U1547B'),
         site == 'U1547B' | (site == 'U1545B' & `Depth (mbsf)` <= 66)) |>
  mutate(S04_CH4_transition_zone = if_else(`SO42- (mM)` != 0 & `CH4 (mM)` != 0, `Depth (mbsf)`, NA_real_)) |>
  group_by(site) |>
  mutate(S04_CH4_transition_zone = case_when(
    is.na(S04_CH4_transition_zone) ~ min(S04_CH4_transition_zone[!is.na(S04_CH4_transition_zone)]),
    !is.na(S04_CH4_transition_zone) ~ min(S04_CH4_transition_zone[!is.na(S04_CH4_transition_zone)]))) |>
  relocate(S04_CH4_transition_zone, .after = 3) |>

  pivot_longer(
    cols = 5:last_col(),
    names_to = 'parameter',
    values_to = 'measurement') |>
  mutate(
    parameter = if_else(parameter == 'SigmaH2S (uM)', 'H2S (uM)', parameter),
    parameter = parameter |> factor(levels = c(
      'T', 'DOC (mM)', 'DIC (mM)', 'CH4 (mM)', 'SO42- (mM)',
      'H2S (uM)', 'H2 (nM)', 'CO (nM)', 'NH4+ (mM)')),
    color_code = if_else(measurement == 0, 'zero', 'non-zero'))


env_params_1546_1547_comparison_plot =
  env_params_1546_1547_comparison_data |>
  ggplot(aes(measurement, `Depth (mbsf)`, color = color_code)) +
  geom_point(size = 1) +
  facet_grid(vars(site), vars(parameter), scales = 'free_x') +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle = 55, hjust = 1),
        legend.position = 'top') +
  scale_y_reverse() +
  labs(
    y = 'Depth (mbsf)\n',
    x = '\nMeasurement',
    color = 'Concentration'
  )
#geom_hline(aes(yintercept = S04_CH4_transition_zone))

env_params_1546_1547_comparison_plot

```

<br>

## Run Trimmomatic on metagenomes
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=compute # Queue selection
#SBATCH --job-name=imo # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/guaymas_andreas_metagenomes_trimmomatic_fastqc_nov_2022_%A-%a.log # Standard output/error
#SBATCH --array=0-12
#SBATCH --qos=unlim

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/Guaymas_Basin_Subsurface

module load anaconda/5.1
source activate trimmomatic

ADAPTER_FASTA=$(echo "/vortexfs1/home/dgellermcgrath/.conda/envs/trimmomatic/share/trimmomatic-0.39-2/adapters/NexteraPE-PE.fa")

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/Guaymas_Basin_Subsurface")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas")

FASTQC_OUTDIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/fastqc_of_trimmed_andreas_read_files")

FILE_PATTERN=("1545B_13H_4_S5" "1545B_19F_3_S6" "1545B_1H_2_S1" "1545B_2H_3_S2" "1545B_32F_3_S7" "1545B_4_H3_S3" "1545B_8_H3_S4" "1547B_1H_3_S8" "1547B_2H_3_S9" "1547B_3H_3_S10" "1547B_5H_3_S11" "1547B_7H_3_S12" "1547B_9H_3_S13")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


trimmomatic PE \
-threads 36 \
-trimlog ${OUT_DIR}/trim_logs/${FILE_PATTERN}.log \
-summary ${OUT_DIR}/trim_summaries/${FILE_PATTERN}_statsSummaryFile.txt \
${IN_DIR}/${FILE_PATTERN}_R1_001.fastq.gz \
${IN_DIR}/${FILE_PATTERN}_R2_001.fastq.gz \
${OUT_DIR}/paired_output/${FILE_PATTERN}_paired_1.fastq.gz \
${OUT_DIR}/unpaired_output/${FILE_PATTERN}_unpaired_1.fastq.gz \
${OUT_DIR}/paired_output/${FILE_PATTERN}_paired_2.fastq.gz \
${OUT_DIR}/unpaired_output/${FILE_PATTERN}_unpaired_2.fastq.gz \
ILLUMINACLIP:${ADAPTER_FASTA}:2:30:10 \
LEADING:20 \
TRAILING:20 \
SLIDINGWINDOW:04:24 \
MINLEN:50


fastqc \
-t 35 \
-o ${FASTQC_OUTDIR}/ \
${OUT_DIR}/paired_output/${FILE_PATTERN}*.fastq.gz
```

<br>

## Run FastQC on metagenomes after read quality trimming with Trimmomatic
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=compute # Queue selection
#SBATCH --job-name=imo # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/guaymas_andreas_metagenomes_trimmomatic_fastqc_nov_2022_%A-%a.log # Standard output/error
#SBATCH --array=0-12
#SBATCH --qos=unlim

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/Guaymas_Basin_Subsurface

module load anaconda/5.1
source activate trimmomatic

ADAPTER_FASTA=$(echo "/vortexfs1/home/dgellermcgrath/.conda/envs/trimmomatic/share/trimmomatic-0.39-2/adapters/NexteraPE-PE.fa")

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/Guaymas_Basin_Subsurface")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas")

FASTQC_OUTDIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/fastqc_of_trimmed_andreas_read_files")

FILE_PATTERN=("1545B_13H_4_S5" "1545B_19F_3_S6" "1545B_1H_2_S1" "1545B_2H_3_S2" "1545B_32F_3_S7" "1545B_4_H3_S3" "1545B_8_H3_S4" "1547B_1H_3_S8" "1547B_2H_3_S9" "1547B_3H_3_S10" "1547B_5H_3_S11" "1547B_7H_3_S12" "1547B_9H_3_S13")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


fastqc \
-t 35 \
-o ${FASTQC_OUTDIR}/ \
${OUT_DIR}/paired_output/${FILE_PATTERN}*.fastq.gz
```

<br>

## Run FastX and FastQC on metagenomes after initial FastQC; trim low quality ends
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=fastx_qc # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/guaymas_metagenomes_fastx_and_fastqc_%A-%a.log # Standard output/error
#SBATCH --array=0-25
#SBATCH --qos=scavenger

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas

module load anaconda/5.1
source activate trimmomatic


IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas/paired_output")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/trimmed_fastq_andreas/fastx_final_trimmed")

FASTQC_OUTDIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/andreas_metagenomes/fastqc_of_trimmed_andreas_read_files/final_fastqc_after_fastx_trimming_nov_14_2022")

FILE_PATTERN=("1545B_13H_4_S5_paired_1.fastq" "1545B_13H_4_S5_paired_2.fastq" "1545B_19F_3_S6_paired_1.fastq" "1545B_19F_3_S6_paired_2.fastq" "1545B_1H_2_S1_paired_1.fastq" "1545B_1H_2_S1_paired_2.fastq" "1545B_2H_3_S2_paired_1.fastq" "1545B_2H_3_S2_paired_2.fastq" "1545B_32F_3_S7_paired_1.fastq" "1545B_32F_3_S7_paired_2.fastq" "1545B_4_H3_S3_paired_1.fastq" "1545B_4_H3_S3_paired_2.fastq" "1545B_8_H3_S4_paired_1.fastq" "1545B_8_H3_S4_paired_2.fastq" "1547B_1H_3_S8_paired_1.fastq" "1547B_1H_3_S8_paired_2.fastq" "1547B_2H_3_S9_paired_1.fastq" "1547B_2H_3_S9_paired_2.fastq" "1547B_3H_3_S10_paired_1.fastq" "1547B_3H_3_S10_paired_2.fastq" "1547B_5H_3_S11_paired_1.fastq" "1547B_5H_3_S11_paired_2.fastq" "1547B_7H_3_S12_paired_1.fastq" "1547B_7H_3_S12_paired_2.fastq" "1547B_9H_3_S13_paired_1.fastq" "1547B_9H_3_S13_paired_2.fastq")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}

#gunzip trimmed fastq files prior to fastx trimming; it doesn't work on gzipped files
# cd $IN_DIR
# gunzip *.gz

fastx_trimmer \
-f 15 \
-i "$IN_DIR"/"$FILE_PATTERN" \
-o "$OUT_DIR"/"$FILE_PATTERN"

pigz "$OUT_DIR"/"$FILE_PATTERN"

fastqc \
-t 35 \
-o "$FASTQC_OUTDIR"/ \
"$OUT_DIR"/"$FILE_PATTERN".gz
```

<br>

## Co-assemble 26 Guaymas basin drill core metagenomes using MEGAHIT
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=bigmem # Queue selection
#SBATCH --job-name=co # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=79 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=3000gb # Job memory request
#SBATCH --time=10-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/26_guaymas_metagenomes_assembly_coassembly_megahit_default_nov_14_2022_%j.log # Standard output/error
#SBATCH --qos=unlim

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/all_fastx_trimmed_reads
source activate megahit

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly")

megahit -t 79 -1 1545B_13H_4_S5_paired_1.fastq.gz,1545B_19F_3_S6_paired_1.fastq.gz,1545B_1H_2_S1_paired_1.fastq.gz,1545B_2H_3_S2_paired_1.fastq.gz,1545B_32F_3_S7_paired_1.fastq.gz,1545B_4_H3_S3_paired_1.fastq.gz,1545B_8_H3_S4_paired_1.fastq.gz,1546B-1H2_S1_L004_paired_1.fastq.gz,1546B-3H2_S2_L004_paired_1.fastq.gz,1547B-1H2_S3_L004_paired_1.fastq.gz,1547B_1H_3_S8_paired_1.fastq.gz,1547B_2H_3_S9_paired_1.fastq.gz,1547B_3H_3_S10_paired_1.fastq.gz,1547B_5H_3_S11_paired_1.fastq.gz,1547B_7H_3_S12_paired_1.fastq.gz,1547B-8H2_S4_L004_paired_1.fastq.gz,1547B_9H_3_S13_paired_1.fastq.gz,1548B-2H3_S5_L004_paired_1.fastq.gz,1548B-4H7_S6_L004_paired_1.fastq.gz,1548B-8H5_S7_L004_paired_1.fastq.gz,1549B-3H2_S8_L004_paired_1.fastq.gz,U1545B_34F3_S9_L004_paired_1.fastq.gz,U1545B_4H2_S18_L004_paired_1.fastq.gz,U1547B_2H2_S12_L004_paired_1.fastq.gz,U1547B_5H2_S11_L004_paired_1.fastq.gz,U1547B_9H2_S14_L004_paired_1.fastq.gz -2 1545B_13H_4_S5_paired_2.fastq.gz,1545B_19F_3_S6_paired_2.fastq.gz,1545B_1H_2_S1_paired_2.fastq.gz,1545B_2H_3_S2_paired_2.fastq.gz,1545B_32F_3_S7_paired_2.fastq.gz,1545B_4_H3_S3_paired_2.fastq.gz,1545B_8_H3_S4_paired_2.fastq.gz,1546B-1H2_S1_L004_paired_2.fastq.gz,1546B-3H2_S2_L004_paired_2.fastq.gz,1547B-1H2_S3_L004_paired_2.fastq.gz,1547B_1H_3_S8_paired_2.fastq.gz,1547B_2H_3_S9_paired_2.fastq.gz,1547B_3H_3_S10_paired_2.fastq.gz,1547B_5H_3_S11_paired_2.fastq.gz,1547B_7H_3_S12_paired_2.fastq.gz,1547B-8H2_S4_L004_paired_2.fastq.gz,1547B_9H_3_S13_paired_2.fastq.gz,1548B-2H3_S5_L004_paired_2.fastq.gz,1548B-4H7_S6_L004_paired_2.fastq.gz,1548B-8H5_S7_L004_paired_2.fastq.gz,1549B-3H2_S8_L004_paired_2.fastq.gz,U1545B_34F3_S9_L004_paired_2.fastq.gz,U1545B_4H2_S18_L004_paired_2.fastq.gz,U1547B_2H2_S12_L004_paired_2.fastq.gz,U1547B_5H2_S11_L004_paired_2.fastq.gz,U1547B_9H2_S14_L004_paired_2.fastq.gz -o ${OUT_DIR}
```

<br>

## Create bins from the metagenomic co-assembly using MaxBin2
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=compute # Queue selection
#SBATCH --job-name=mxbc # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=35 #
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=7-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/maxbin2_andreas_coassembly_binning_nov_26_2022_%j.log# Standard output/error
#SBATCH --qos=unlim

source activate maxbin2_final

# create bins with maxbin2 -- on the original megabit coassembly

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly")

BAM_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/bam_files")

BIN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/maxbin2")

run_MaxBin.pl \
-contig  "$IN_DIR"/final.contigs.fa \
-abund "$BAM_DIR"/andreas_coassembly_contig_coverage_from_default_coverM_settings_nov_25_2022.txt \
-out "$BIN_DIR" \
-thread 35 \
-markerset 40
```

<br>

## Create bins from the metagenomic co-assembly using MetaBat2
```{bash, eval=FALSE, include=TRUE}
conda activate metabat2

#command to create coverage file from bam files for metabat2
jgi_summarize_bam_contig_depths --outputDepth summaryFile.txt *.bam 

# mv summaryFile.txt andreas_coassembly_contig_coverage_from_default_coverM_settings_nov_25_2022.txt


# create bins with metabat2 -- on the original metabat2 coassembly - made with default megahit settings

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly")

BAM_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/bam_files")

BIN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/metabat2")

metabat2 \
--saveCls \
--numThreads 36 \
-i  "$IN_DIR"/final.contigs.fa \
-a "$BAM_DIR"/andreas_coassembly_contig_coverage_from_default_coverM_settings_nov_25_2022.txt \
-o "$BIN_DIR"/MAG_ \
--unbinned

```

<br>

## Create bins from metagenomic co-assembly using CONCOCT
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=samtools # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 #
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/samtools_sort_and_index_array_%A-%a.log# Standard output/error
#SBATCH --array=0-25
#SBATCH --qos=scavenger

# command to print out patterns used in FILE_PATTERN array below
# ls *.bam | sed -r "s/.*.fa.(.*).fastq.*/\"\1\"/" | tr '\n' ' '

source activate concoct

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/nov28_bam_files

FILE_PATTERN=("1545B_13H_4_S5_paired_1" "1545B_19F_3_S6_paired_1" "1545B_1H_2_S1_paired_1" "1545B_2H_3_S2_paired_1" "1545B_32F_3_S7_paired_1" "1545B_4_H3_S3_paired_1" "1545B_8_H3_S4_paired_1" "1546B-1H2_S1_L004_paired_1" "1546B-3H2_S2_L004_paired_1" "1547B-1H2_S3_L004_paired_1" "1547B_1H_3_S8_paired_1" "1547B_2H_3_S9_paired_1" "1547B_3H_3_S10_paired_1" "1547B_5H_3_S11_paired_1" "1547B_7H_3_S12_paired_1" "1547B-8H2_S4_L004_paired_1" "1547B_9H_3_S13_paired_1" "1548B-2H3_S5_L004_paired_1" "1548B-4H7_S6_L004_paired_1" "1548B-8H5_S7_L004_paired_1" "1549B-3H2_S8_L004_paired_1" "U1545B_34F3_S9_L004_paired_1" "U1545B_4H2_S18_L004_paired_1" "U1547B_2H2_S12_L004_paired_1" "U1547B_5H2_S11_L004_paired_1" "U1547B_9H2_S14_L004_paired_1")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


# create sorted bam file
samtools sort final.contigs.fa."$FILE_PATTERN".fastq.gz.bam -o final.contigs.fa."$FILE_PATTERN".fastq.gz.sorted.bam


# create indexed bam file
samtools index final.contigs.fa."$FILE_PATTERN".fastq.gz.sorted.bam final.contigs.fa."$FILE_PATTERN".fastq.gz.sorted.bam.bai

-----
-----
# done interactively on a compute node:

conda activate concoct

BAM_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/nov28_bam_files")

CONCOCT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/concoct")

# cut up contigs into smaller parts
cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly
cut_up_fasta.py final.contigs.fa -c 10000 -o 0 --merge_last -b contigs_10K.bed > contigs_10K.fa

# Generate table with coverage depth information per sample and subcontig. This step assumes the directory 'mapping' contains sorted and indexed bam files where each sample has been mapped against the original contigs.

conda deactivate
conda activate concoct

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly
concoct_coverage_table.py contigs_10K.bed "$BAM_DIR"/*.sorted.bam > "$CONCOCT_DIR"/metadata/nov28_concoct_coverage_table.tsv

# run concoct
NUMEXPR_MAX_THREADS=35
concoct --composition_file contigs_10K.fa --coverage_file "$CONCOCT_DIR"/metadata/nov28_concoct_coverage_table.tsv -b "$CONCOCT_DIR"/ -t 35

# Merge subcontig clustering into original contig clustering
merge_cutup_clustering.py "$CONCOCT_DIR"/clustering_gt1000.csv > "$CONCOCT_DIR"/clustering_merged.csv

# Extract bins as individual FASTA
mkdir "$CONCOCT_DIR"/fasta_bins
extract_fasta_bins.py final.contigs.fa "$CONCOCT_DIR"/clustering_merged.csv --output_path "$CONCOCT_DIR"/fasta_bins
```

<br>

## Aggregate and dereplicate bins using Das Tool
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=dasTool # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=35 #
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=4-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/das_tool_nov30_2022_%j.log# Standard output/error
#SBATCH --qos=scavenger

# following commands were run interactively on a compute node:

#conda activate das_tool

#Fasta_to_Contig2Bin.sh -e fa -i /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/concoct/fasta_bins > concoct_contig2bin.tsv

# truncated contig headers & contig2bin contig headers; was getting error with DAS Tool

# cp final.contigs.fa truncatedHeaders_copy_final.contigs.faq

#cat truncatedHeaders_copy_final.contigs.fa | sed -r 's/(^>.*) flag.*/\1/' > tmp.fa && mv -f tmp.fa truncatedHeaders_copy_final.contigs.fa

#cat concoct_contig2bin.tsv | sed -r 's/(^k.*) flag.*len=[0-9]+\s+([0-9]+)/\1\t\2/' > tmp.tsv && mv -f tmp.tsv concoct_contig2bin.tsv

#not changing metabat2 and maxbin2 contig2bin files. by default it has kept the k_[0-9]+ part of the megahit coassembly contig headers, and gotten rid of everything after the first space. that's probably why DAS Tool was failing earlier...

#Fasta_to_Contig2Bin.sh -i /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/metabat2 -e fna > metabat2_contig2bin.tsv

#Fasta_to_Contig2Bin.sh -i /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/maxbin2 -e fasta > maxbin2_contig2bin.tsv

source activate das_tool

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/dasTool_analysis/contig2bin_inputs

# create refined bins with DAS_tool

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/megahit_coassembly")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/dasTool_analysis/final_refined_bins_november_2022")


DAS_Tool \
-i concoct_contig2bin.tsv,metabat2_contig2bin.tsv,maxbin2_contig2bin.tsv \
-l concoct,metabat,maxbin \
-c "$IN_DIR"/truncatedHeaders_copy_final.contigs.fa \
-o "$OUT_DIR"/Guaymas_MAG_P_ \
--write_bins \
-t 35
```

<br>

## Classify taxonomy of Das Tool bins
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=gtdb-tk # taxonomic assessent of DAS Tool Guaymas bins
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=35 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=187gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/andreas_coassembly_gtdbtk_on_guaymasDasToolMegahitBins_%j.log # Standard output/error
#SBATCH --qos=scavenger

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385

source activate gtdbtk

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/gtdbtk")

gtdbtk classify_wf \
--cpus 35 \
--pplacer_cpus 1 \
--genome_dir "$IN_DIR" \
-x fa \
--force \
--out_dir "$OUT_DIR"
```

<br>

## Run CoverM on MAGs to calculate their relative abundances in all metagenomic samples
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=coverM # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=35 #
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/dec_1_2022_andreas_dasTool_mags_relAbun_coverM_array_%A-%a.log# Standard output/error
#SBATCH --array=0-25
#SBATCH --qos=scavenger

source activate coverm

cd /vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/all_fastx_trimmed_reads

FILE_PATTERN=("1545B_13H_4_S5" "1545B_19F_3_S6" "1545B_1H_2_S1" "1545B_2H_3_S2" "1545B_32F_3_S7" "1545B_4_H3_S3" "1545B_8_H3_S4" "1546B-1H2_S1_L004" "1546B-3H2_S2_L004" "1547B-1H2_S3_L004" "1547B_1H_3_S8" "1547B_2H_3_S9" "1547B_3H_3_S10" "1547B_5H_3_S11" "1547B_7H_3_S12" "1547B-8H2_S4_L004" "1547B_9H_3_S13" "1548B-2H3_S5_L004" "1548B-4H7_S6_L004" "1548B-8H5_S7_L004" "1549B-3H2_S8_L004" "U1545B_34F3_S9_L004" "U1545B_4H2_S18_L004" "U1547B_2H2_S12_L004" "U1547B_5H2_S11_L004" "U1547B_9H2_S14_L004")

GENOME_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/rel_abun_files")


FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}

# always double check whether fastq files are gzipped or not

coverm genome \
--coupled "$FILE_PATTERN"_paired_1.fastq.gz "$FILE_PATTERN"_paired_2.fastq.gz \
--genome-fasta-directory "$GENOME_DIR" \
--mapper bwa-mem \
--methods relative_abundance \
--min-read-percent-identity 95 \
--min-read-aligned-percent 50 \
--min-covered-fraction 0 \
--exclude-supplementary \
--genome-fasta-extension fa \
--threads 35 \
--output-file "$OUT_DIR"/"$FILE_PATTERN"_refined_bin_coverages.tsv
```

<br>

## Call translated genes with Prodigal; annotate .faa output files with kofamKOALA command line tool (KofamScan)
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=pr_kf # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=187gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/dec_5_2022_prodigal_kofamscan_on_guaymas_mags_array_%A-%a.log# Standard output/error
#SBATCH --array=0-141
#SBATCH --qos=scavenger


# load conda environment containing Prodigal and Kofamscan
source activate upd-kofam

# kofamscan hmm profiles directory
HMM_DIR=$(echo "/vortexfs1/omics/pachiadaki/dgellermcgrath/kofam_hmm_profiles")

# input directories
MAG_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")

# output directories
PROD_FNA_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/prodigal/fna")
PROD_FAA_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/prodigal/faa")
KOFAM_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/kofamscan")


#mag name file pattern
FILE_PATTERN=("MAG_130" "MAG_51" "MAG_244" "MAG_263" "MAG_12" "MAG_229" "145" "MAG_292" "MAG_117" "174" "112_sub" "MAG_305" "MAG_284" "MAG_334" "26_sub" "MAG_119" "280" "MAG_33" "MAG_116" "MAG_53" "MAG_249" "MAG_210" "MAG_315" "289" "MAG_155" "MAG_46" "MAG_153_sub" "MAG_144" "MAG_234" "MAG_200" "73_sub" "MAG_162" "MAG_193" "MAG_126" "275_sub" "MAG_328" "MAG_141" "259_sub" "MAG_313" "MAG_149_sub" "MAG_214" "MAG_289" "MAG_133" "MAG_247" "MAG_273_sub" "MAG_206" "16_sub" "MAG_128" "MAG_280_sub" "MAG_56_sub" "MAG_39" "MAG_226" "MAG_140" "MAG_172" "270_sub" "MAG_142" "MAG_327_sub" "MAG_326" "2" "MAG_42_sub" "MAG_94" "58" "131_sub" "245_sub" "238_sub" "MAG_13" "MAG_332_sub" "97_sub" "MAG_171" "163" "MAG_277" "MAG_235" "MAG_80_sub" "MAG_58" "MAG_18" "MAG_199" "MAG_114" "MAG_222_sub" "MAG_269_sub" "123_sub" "MAG_121" "169_sub" "MAG_217_sub" "MAG_74" "MAG_316" "MAG_224" "103_sub" "MAG_314" "223_sub")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


prodigal \
-i "$MAG_DIR"/"$FILE_PATTERN".fa \
-a "$PROD_FAA_DIR"/"$FILE_PATTERN".faa \
-d "$PROD_FNA_DIR"/"$FILE_PATTERN".fna


exec_annotation \
-o "$KOFAM_DIR"/"$FILE_PATTERN"-ko.tsv \
--cpu 36 \
--tmp-dir "$KOFAM_DIR"/"$FILE_PATTERN" \
-f detail-tsv \
-p "$HMM_DIR"/profiles/prokaryote.hal \
-k "$HMM_DIR"/ko_list "$PROD_FAA_DIR"/"$FILE_PATTERN".faa
```

<br>

## Annotate the MAGs using Prokka
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=scavenger # Queue selection
#SBATCH --job-name=prokka # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=187gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/feb_3_2023_prokka_annotation_of_guaymas_mags_array_%A-%a.log# Standard output/error
#SBATCH --array=0-88
#SBATCH --qos=scavenger


# directory containing the Guaymas MAGs
MAG_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")


# directory for all dram annotation output
OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/prokka/annotations")


# load prokka conda environment
source activate prokka


# bash array of mag name identifiers
FILE_PATTERN=("MAG_130" "MAG_51" "MAG_244" "MAG_263" "MAG_12" "MAG_229" "145" "MAG_292" "MAG_117" "174" "112_sub" "MAG_305" "MAG_284" "MAG_334" "26_sub" "MAG_119" "280" "MAG_33" "MAG_116" "MAG_53" "MAG_249" "MAG_210" "MAG_315" "289" "MAG_155" "MAG_46" "MAG_153_sub" "MAG_144" "MAG_234" "MAG_200" "73_sub" "MAG_162" "MAG_193" "MAG_126" "275_sub" "MAG_328" "MAG_141" "259_sub" "MAG_313" "MAG_149_sub" "MAG_214" "MAG_289" "MAG_133" "MAG_247" "MAG_273_sub" "MAG_206" "16_sub" "MAG_128" "MAG_280_sub" "MAG_56_sub" "MAG_39" "MAG_226" "MAG_140" "MAG_172" "270_sub" "MAG_142" "MAG_327_sub" "MAG_326" "2" "MAG_42_sub" "MAG_94" "58" "131_sub" "245_sub" "238_sub" "MAG_13" "MAG_332_sub" "97_sub" "MAG_171" "163" "MAG_277" "MAG_235" "MAG_80_sub" "MAG_58" "MAG_18" "MAG_199" "MAG_114" "MAG_222_sub" "MAG_269_sub" "123_sub" "MAG_121" "169_sub" "MAG_217_sub" "MAG_74" "MAG_316" "MAG_224" "103_sub" "MAG_314" "223_sub")


# pick one mag name for the current array iteration
FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


# run prokka
prokka \
--outdir "$OUT_DIR"/"$FILE_PATTERN" \
--prefix "$FILE_PATTERN" \
--cpus 35 \
"$MAG_DIR"/"$FILE_PATTERN".fa
```

<br>

## Map metatranscriptomic reads to the MAGs
```{bash, eval=FALSE, include=TRUE}
#!/bin/bash
#SBATCH --partition=compute # Queue selection
#SBATCH --job-name=coverM # Job name
#SBATCH --mail-type=ALL # Mail events (BEGIN, END, FAIL, ALL)
#SBATCH --mail-user=dgellermcgrath@gmail.com # Where to send mail
#SBATCH --ntasks=36 # Run on one CPU
#SBATCH --nodes=1
#SBATCH --cpus-per-task=1
#SBATCH --mem=180gb # Job memory request
#SBATCH --time=1-00:00:00 # Time limit hrs:min:sec
#SBATCH --output=logs/feb_4_2023_coverm_mag_counts_of_metaT_reads_mapped_array_%A-%a.log# Standard output/error
#SBATCH --array=0-18
#SBATCH --qos=unlim

source activate coverm

IN_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/transcriptomes/Fastx_output")

MAG_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/bins/final_bins_dec_2022")

OUT_DIR=$(echo "/vortexfs1/omics/edgcomb/home_vedgcomb/Beaudoin/IODP_385/metagenomes/FINAL_GUAYMAS_ANALYSIS_NOV_2022/coverM/metaT/reads_mapped_per_genome")

FILE_PATTERN=("1545_11H3" "1545_1H2" "1545_4H3" "1546_12H2" "1546_1H2" "1546_3H2" "1547_1H2" "1547_5H2" "1547_9H2" "1548_1H2" "1548_4H7" "1549_1H2" "1549_3H2" "1550_1H2" "1550_3H2" "1551_1H2" "1551_3H2" "1552_1H2" "1552_3H4")

FILE_PATTERN=${FILE_PATTERN[$SLURM_ARRAY_TASK_ID]}


coverm \
genome \
--coupled "$IN_DIR"/"$FILE_PATTERN"_R1_paired_fastx.fastq "$IN_DIR"/"$FILE_PATTERN"_R2_paired_fastx.fastq \
--genome-fasta-directory "$MAG_DIR" \
--genome-fasta-extension fa \
--mapper bwa-mem \
--methods count relative_abundance mean trimmed_mean rpkm tpm \
--min-covered-fraction 0 \
--min-read-percent-identity 95 \
--min-read-aligned-percent 50 \
--exclude-supplementary \
--threads 36 \
-o "$OUT_DIR"/"$FILE_PATTERN"_counts_reads_mapped_per_genome.tsv
```
