# Metagenomic Profiles of Archaea and Bacteria within Thermal and Geochemical Gradients of the Guaymas Basin Deep Subsurface

[![DOI](https://zenodo.org/badge/601356394.svg)](https://zenodo.org/badge/latestdoi/601356394)

This repository contains scripts, data, and computing environments used for the analysis of metagenomic and metatranscriptomic datasets from the Guaymas Basin, Mexico for the study "Metagenomic Profiles of Archaea and Bacteria within Thermal and Geochemical Gradients of the Guaymas Basin Deep Subsurface". The authors of this manuscript are D. Geller-McGrath, P. Mara, V. Edgcomb, D. Beaudoin, Y. Morono, and A. Teske. The scripts, files, and data here were written and uploaded by D. Geller-McGrath.

To view the code used for this analysis, click here: https://d-mcgrath.github.io/guaymas_basin/. This is a GitHub pages document that contains R scripts for figure creation and data analysis, as well as bash scripts for data analysis that was done using a high performance computing cluster (HPC). The R script chunks contain in order the code for creation of figures 2-6 in the main text, and supplementary figures 2-4. Bash scripts for analysis done on an HPC are below the R scripts. Note that some figures were aesthetically adjusted using Adobe Illustrator. Figure 1 and Supplementary Figure 1 were created using Excel.

The KofamScan annotations were used to assign KEGG annotations to KEGG modules to give broad overview of the metabolisms present in the genomes recovered from Guaymas Basin. The associated script was used to generate Supplementary Figures 3 and 4, and Supplementary Data 5. To present additional data on Chloroflexota, Thermoproteota, Acidobacteriota, Desulfobacterota, Aerophobota, and WOR-3 in a user-friendly format, METABOLIC was used to annotate MAGs to identify putatative metabolisms that we predicted would be present in Guaymas Basin. The associated script was used to generate Supplementary Data 6b. It is recognized that different databases used by the different tools can provide slightly different information.
